diff --git a/build/config/sysroot.gni b/build/config/sysroot.gni
index f26a55f..a3003c8 100644
--- a/build/config/sysroot.gni
+++ b/build/config/sysroot.gni
@@ -17,7 +17,7 @@ declare_args() {
 
   use_sysroot =
       current_cpu == "x86" || current_cpu == "x64" || current_cpu == "arm" ||
-      current_cpu == "arm64" || current_cpu == "mipsel"
+      current_cpu == "arm64" || current_cpu == "mipsel" || current_cpu == "mips64el"
 }
 
 if (current_os == target_os && current_cpu == target_cpu &&
@@ -49,6 +49,8 @@ if (current_os == target_os && current_cpu == target_cpu &&
     sysroot = "$target_sysroot_dir/debian_stretch_i386-sysroot"
   } else if (current_cpu == "mipsel") {
     sysroot = "$target_sysroot_dir/debian_stretch_mips-sysroot"
+  } else if (current_cpu == "mips64el") {
+    sysroot = "$target_sysroot_dir/debian_jessie_mips64-sysroot"
   } else if (current_cpu == "arm") {
     sysroot = "$target_sysroot_dir/debian_stretch_arm-sysroot"
   } else if (current_cpu == "arm64") {
