95ecb913508e3dfc32afb93f6b930dc8d8514de9
diff --git a/build/linux/sysroot_scripts/packagelist.jessie.amd64 b/build/linux/sysroot_scripts/packagelist.jessie.amd64
index 38cdef746f0c..7642c264b57a 100644
--- a/build/linux/sysroot_scripts/packagelist.jessie.amd64
+++ b/build/linux/sysroot_scripts/packagelist.jessie.amd64
@@ -10,8 +10,6 @@ http://ftp.us.debian.org/debian/pool/main/a/attr/libattr1_2.4.47-2_amd64.deb
 http://ftp.us.debian.org/debian/pool/main/a/audit/libaudit1_2.4-1+b1_amd64.deb
 http://ftp.us.debian.org/debian/pool/main/a/avahi/libavahi-client3_0.6.31-5_amd64.deb
 http://ftp.us.debian.org/debian/pool/main/a/avahi/libavahi-common3_0.6.31-5_amd64.deb
-http://ftp.us.debian.org/debian/pool/main/b/bluez/libbluetooth3_5.23-2+b1_amd64.deb
-http://ftp.us.debian.org/debian/pool/main/b/bluez/libbluetooth-dev_5.23-2+b1_amd64.deb
 http://ftp.us.debian.org/debian/pool/main/b/brltty/libbrlapi0.6_5.2~20141018-5_amd64.deb
 http://ftp.us.debian.org/debian/pool/main/b/brltty/libbrlapi-dev_5.2~20141018-5_amd64.deb
 http://ftp.us.debian.org/debian/pool/main/c/cairo/libcairo2_1.14.0-2.1+deb8u2_amd64.deb
@@ -21,6 +19,8 @@ http://ftp.us.debian.org/debian/pool/main/c/cairo/libcairo-script-interpreter2_1
 http://ftp.us.debian.org/debian/pool/main/d/dbus-glib/libdbus-glib-1-2_0.102-1_amd64.deb
 http://ftp.us.debian.org/debian/pool/main/d/dbus/libdbus-1-3_1.8.22-0+deb8u1_amd64.deb
 http://ftp.us.debian.org/debian/pool/main/d/dbus/libdbus-1-dev_1.8.22-0+deb8u1_amd64.deb
+http://ftp.us.debian.org/debian/pool/main/d/d-conf/libdconf1_0.22.0-1_amd64.deb
+http://ftp.us.debian.org/debian/pool/main/d/d-conf/libdconf-dev_0.22.0-1_amd64.deb
 http://ftp.us.debian.org/debian/pool/main/e/e2fsprogs/comerr-dev_2.1-1.42.12-2+b1_amd64.deb
 http://ftp.us.debian.org/debian/pool/main/e/e2fsprogs/libcomerr2_1.42.12-2+b1_amd64.deb
 http://ftp.us.debian.org/debian/pool/main/e/elfutils/libelf1_0.159-4.2_amd64.deb
@@ -169,6 +169,8 @@ http://ftp.us.debian.org/debian/pool/main/x/x11proto-xext/x11proto-xext-dev_7.3.
 http://ftp.us.debian.org/debian/pool/main/x/x11proto-xinerama/x11proto-xinerama-dev_1.2.1-2_all.deb
 http://ftp.us.debian.org/debian/pool/main/z/zlib/zlib1g_1.2.8.dfsg-2+b1_amd64.deb
 http://ftp.us.debian.org/debian/pool/main/z/zlib/zlib1g-dev_1.2.8.dfsg-2+b1_amd64.deb
+http://security.debian.org/pool/updates/main/b/bluez/libbluetooth3_5.23-2+deb8u1_amd64.deb
+http://security.debian.org/pool/updates/main/b/bluez/libbluetooth-dev_5.23-2+deb8u1_amd64.deb
 http://security.debian.org/pool/updates/main/c/cups/libcups2_1.7.5-11+deb8u1_amd64.deb
 http://security.debian.org/pool/updates/main/c/cups/libcups2-dev_1.7.5-11+deb8u1_amd64.deb
 http://security.debian.org/pool/updates/main/e/expat/libexpat1_2.1.0-6+deb8u4_amd64.deb
@@ -177,8 +179,8 @@ http://security.debian.org/pool/updates/main/f/fontconfig/libfontconfig1_2.11.0-
 http://security.debian.org/pool/updates/main/f/fontconfig/libfontconfig1-dev_2.11.0-6.3+deb8u1_amd64.deb
 http://security.debian.org/pool/updates/main/f/freetype/libfreetype6_2.5.2-3+deb8u2_amd64.deb
 http://security.debian.org/pool/updates/main/f/freetype/libfreetype6-dev_2.5.2-3+deb8u2_amd64.deb
-http://security.debian.org/pool/updates/main/g/gdk-pixbuf/libgdk-pixbuf2.0-0_2.31.1-2+deb8u5_amd64.deb
-http://security.debian.org/pool/updates/main/g/gdk-pixbuf/libgdk-pixbuf2.0-dev_2.31.1-2+deb8u5_amd64.deb
+http://security.debian.org/pool/updates/main/g/gdk-pixbuf/libgdk-pixbuf2.0-0_2.31.1-2+deb8u6_amd64.deb
+http://security.debian.org/pool/updates/main/g/gdk-pixbuf/libgdk-pixbuf2.0-dev_2.31.1-2+deb8u6_amd64.deb
 http://security.debian.org/pool/updates/main/g/glibc/libc6_2.19-18+deb8u10_amd64.deb
 http://security.debian.org/pool/updates/main/g/glibc/libc6-dev_2.19-18+deb8u10_amd64.deb
 http://security.debian.org/pool/updates/main/g/gnutls28/libgnutls28-dev_3.3.8-6+deb8u6_amd64.deb
diff --git a/build/linux/sysroot_scripts/packagelist.jessie.arm b/build/linux/sysroot_scripts/packagelist.jessie.arm
index fe81e1c20f8d..4d220fe94852 100644
--- a/build/linux/sysroot_scripts/packagelist.jessie.arm
+++ b/build/linux/sysroot_scripts/packagelist.jessie.arm
@@ -10,8 +10,6 @@ http://ftp.us.debian.org/debian/pool/main/a/attr/libattr1_2.4.47-2_armhf.deb
 http://ftp.us.debian.org/debian/pool/main/a/audit/libaudit1_2.4-1+b1_armhf.deb
 http://ftp.us.debian.org/debian/pool/main/a/avahi/libavahi-client3_0.6.31-5_armhf.deb
 http://ftp.us.debian.org/debian/pool/main/a/avahi/libavahi-common3_0.6.31-5_armhf.deb
-http://ftp.us.debian.org/debian/pool/main/b/bluez/libbluetooth3_5.23-2+b1_armhf.deb
-http://ftp.us.debian.org/debian/pool/main/b/bluez/libbluetooth-dev_5.23-2+b1_armhf.deb
 http://ftp.us.debian.org/debian/pool/main/b/brltty/libbrlapi0.6_5.2~20141018-5_armhf.deb
 http://ftp.us.debian.org/debian/pool/main/b/brltty/libbrlapi-dev_5.2~20141018-5_armhf.deb
 http://ftp.us.debian.org/debian/pool/main/c/cairo/libcairo2_1.14.0-2.1+deb8u2_armhf.deb
@@ -21,6 +19,8 @@ http://ftp.us.debian.org/debian/pool/main/c/cairo/libcairo-script-interpreter2_1
 http://ftp.us.debian.org/debian/pool/main/d/dbus-glib/libdbus-glib-1-2_0.102-1_armhf.deb
 http://ftp.us.debian.org/debian/pool/main/d/dbus/libdbus-1-3_1.8.22-0+deb8u1_armhf.deb
 http://ftp.us.debian.org/debian/pool/main/d/dbus/libdbus-1-dev_1.8.22-0+deb8u1_armhf.deb
+http://ftp.us.debian.org/debian/pool/main/d/d-conf/libdconf1_0.22.0-1_armhf.deb
+http://ftp.us.debian.org/debian/pool/main/d/d-conf/libdconf-dev_0.22.0-1_armhf.deb
 http://ftp.us.debian.org/debian/pool/main/e/e2fsprogs/comerr-dev_2.1-1.42.12-2+b1_armhf.deb
 http://ftp.us.debian.org/debian/pool/main/e/e2fsprogs/libcomerr2_1.42.12-2+b1_armhf.deb
 http://ftp.us.debian.org/debian/pool/main/e/elfutils/libelf1_0.159-4.2_armhf.deb
@@ -166,6 +166,8 @@ http://ftp.us.debian.org/debian/pool/main/x/x11proto-xext/x11proto-xext-dev_7.3.
 http://ftp.us.debian.org/debian/pool/main/x/x11proto-xinerama/x11proto-xinerama-dev_1.2.1-2_all.deb
 http://ftp.us.debian.org/debian/pool/main/z/zlib/zlib1g_1.2.8.dfsg-2+b1_armhf.deb
 http://ftp.us.debian.org/debian/pool/main/z/zlib/zlib1g-dev_1.2.8.dfsg-2+b1_armhf.deb
+http://security.debian.org/pool/updates/main/b/bluez/libbluetooth3_5.23-2+deb8u1_armhf.deb
+http://security.debian.org/pool/updates/main/b/bluez/libbluetooth-dev_5.23-2+deb8u1_armhf.deb
 http://security.debian.org/pool/updates/main/c/cups/libcups2_1.7.5-11+deb8u1_armhf.deb
 http://security.debian.org/pool/updates/main/c/cups/libcups2-dev_1.7.5-11+deb8u1_armhf.deb
 http://security.debian.org/pool/updates/main/e/expat/libexpat1_2.1.0-6+deb8u4_armhf.deb
@@ -174,8 +176,8 @@ http://security.debian.org/pool/updates/main/f/fontconfig/libfontconfig1_2.11.0-
 http://security.debian.org/pool/updates/main/f/fontconfig/libfontconfig1-dev_2.11.0-6.3+deb8u1_armhf.deb
 http://security.debian.org/pool/updates/main/f/freetype/libfreetype6_2.5.2-3+deb8u2_armhf.deb
 http://security.debian.org/pool/updates/main/f/freetype/libfreetype6-dev_2.5.2-3+deb8u2_armhf.deb
-http://security.debian.org/pool/updates/main/g/gdk-pixbuf/libgdk-pixbuf2.0-0_2.31.1-2+deb8u5_armhf.deb
-http://security.debian.org/pool/updates/main/g/gdk-pixbuf/libgdk-pixbuf2.0-dev_2.31.1-2+deb8u5_armhf.deb
+http://security.debian.org/pool/updates/main/g/gdk-pixbuf/libgdk-pixbuf2.0-0_2.31.1-2+deb8u6_armhf.deb
+http://security.debian.org/pool/updates/main/g/gdk-pixbuf/libgdk-pixbuf2.0-dev_2.31.1-2+deb8u6_armhf.deb
 http://security.debian.org/pool/updates/main/g/glibc/libc6_2.19-18+deb8u10_armhf.deb
 http://security.debian.org/pool/updates/main/g/glibc/libc6-dev_2.19-18+deb8u10_armhf.deb
 http://security.debian.org/pool/updates/main/g/gnutls28/libgnutls28-dev_3.3.8-6+deb8u6_armhf.deb
diff --git a/build/linux/sysroot_scripts/packagelist.jessie.arm64 b/build/linux/sysroot_scripts/packagelist.jessie.arm64
index 03cf88d72c17..69da69db9095 100644
--- a/build/linux/sysroot_scripts/packagelist.jessie.arm64
+++ b/build/linux/sysroot_scripts/packagelist.jessie.arm64
@@ -10,8 +10,6 @@ http://ftp.us.debian.org/debian/pool/main/a/attr/libattr1_2.4.47-2_arm64.deb
 http://ftp.us.debian.org/debian/pool/main/a/audit/libaudit1_2.4-1+b1_arm64.deb
 http://ftp.us.debian.org/debian/pool/main/a/avahi/libavahi-client3_0.6.31-5_arm64.deb
 http://ftp.us.debian.org/debian/pool/main/a/avahi/libavahi-common3_0.6.31-5_arm64.deb
-http://ftp.us.debian.org/debian/pool/main/b/bluez/libbluetooth3_5.23-2+b1_arm64.deb
-http://ftp.us.debian.org/debian/pool/main/b/bluez/libbluetooth-dev_5.23-2+b1_arm64.deb
 http://ftp.us.debian.org/debian/pool/main/b/brltty/libbrlapi0.6_5.2~20141018-5_arm64.deb
 http://ftp.us.debian.org/debian/pool/main/b/brltty/libbrlapi-dev_5.2~20141018-5_arm64.deb
 http://ftp.us.debian.org/debian/pool/main/c/cairo/libcairo2_1.14.0-2.1+deb8u2_arm64.deb
@@ -21,6 +19,8 @@ http://ftp.us.debian.org/debian/pool/main/c/cairo/libcairo-script-interpreter2_1
 http://ftp.us.debian.org/debian/pool/main/d/dbus-glib/libdbus-glib-1-2_0.102-1_arm64.deb
 http://ftp.us.debian.org/debian/pool/main/d/dbus/libdbus-1-3_1.8.22-0+deb8u1_arm64.deb
 http://ftp.us.debian.org/debian/pool/main/d/dbus/libdbus-1-dev_1.8.22-0+deb8u1_arm64.deb
+http://ftp.us.debian.org/debian/pool/main/d/d-conf/libdconf1_0.22.0-1_arm64.deb
+http://ftp.us.debian.org/debian/pool/main/d/d-conf/libdconf-dev_0.22.0-1_arm64.deb
 http://ftp.us.debian.org/debian/pool/main/e/e2fsprogs/comerr-dev_2.1-1.42.12-2+b1_arm64.deb
 http://ftp.us.debian.org/debian/pool/main/e/e2fsprogs/libcomerr2_1.42.12-2+b1_arm64.deb
 http://ftp.us.debian.org/debian/pool/main/e/elfutils/libelf1_0.159-4.2_arm64.deb
@@ -167,6 +167,8 @@ http://ftp.us.debian.org/debian/pool/main/x/x11proto-xext/x11proto-xext-dev_7.3.
 http://ftp.us.debian.org/debian/pool/main/x/x11proto-xinerama/x11proto-xinerama-dev_1.2.1-2_all.deb
 http://ftp.us.debian.org/debian/pool/main/z/zlib/zlib1g_1.2.8.dfsg-2+b1_arm64.deb
 http://ftp.us.debian.org/debian/pool/main/z/zlib/zlib1g-dev_1.2.8.dfsg-2+b1_arm64.deb
+http://security.debian.org/pool/updates/main/b/bluez/libbluetooth3_5.23-2+deb8u1_arm64.deb
+http://security.debian.org/pool/updates/main/b/bluez/libbluetooth-dev_5.23-2+deb8u1_arm64.deb
 http://security.debian.org/pool/updates/main/c/cups/libcups2_1.7.5-11+deb8u1_arm64.deb
 http://security.debian.org/pool/updates/main/c/cups/libcups2-dev_1.7.5-11+deb8u1_arm64.deb
 http://security.debian.org/pool/updates/main/e/expat/libexpat1_2.1.0-6+deb8u4_arm64.deb
@@ -175,8 +177,8 @@ http://security.debian.org/pool/updates/main/f/fontconfig/libfontconfig1_2.11.0-
 http://security.debian.org/pool/updates/main/f/fontconfig/libfontconfig1-dev_2.11.0-6.3+deb8u1_arm64.deb
 http://security.debian.org/pool/updates/main/f/freetype/libfreetype6_2.5.2-3+deb8u2_arm64.deb
 http://security.debian.org/pool/updates/main/f/freetype/libfreetype6-dev_2.5.2-3+deb8u2_arm64.deb
-http://security.debian.org/pool/updates/main/g/gdk-pixbuf/libgdk-pixbuf2.0-0_2.31.1-2+deb8u5_arm64.deb
-http://security.debian.org/pool/updates/main/g/gdk-pixbuf/libgdk-pixbuf2.0-dev_2.31.1-2+deb8u5_arm64.deb
+http://security.debian.org/pool/updates/main/g/gdk-pixbuf/libgdk-pixbuf2.0-0_2.31.1-2+deb8u6_arm64.deb
+http://security.debian.org/pool/updates/main/g/gdk-pixbuf/libgdk-pixbuf2.0-dev_2.31.1-2+deb8u6_arm64.deb
 http://security.debian.org/pool/updates/main/g/glibc/libc6_2.19-18+deb8u10_arm64.deb
 http://security.debian.org/pool/updates/main/g/glibc/libc6-dev_2.19-18+deb8u10_arm64.deb
 http://security.debian.org/pool/updates/main/g/gnutls28/libgnutls28-dev_3.3.8-6+deb8u6_arm64.deb
diff --git a/build/linux/sysroot_scripts/packagelist.jessie.i386 b/build/linux/sysroot_scripts/packagelist.jessie.i386
index fa78ac0f44c6..6c8e62351f58 100644
--- a/build/linux/sysroot_scripts/packagelist.jessie.i386
+++ b/build/linux/sysroot_scripts/packagelist.jessie.i386
@@ -10,8 +10,6 @@ http://ftp.us.debian.org/debian/pool/main/a/attr/libattr1_2.4.47-2_i386.deb
 http://ftp.us.debian.org/debian/pool/main/a/audit/libaudit1_2.4-1+b1_i386.deb
 http://ftp.us.debian.org/debian/pool/main/a/avahi/libavahi-client3_0.6.31-5_i386.deb
 http://ftp.us.debian.org/debian/pool/main/a/avahi/libavahi-common3_0.6.31-5_i386.deb
-http://ftp.us.debian.org/debian/pool/main/b/bluez/libbluetooth3_5.23-2+b1_i386.deb
-http://ftp.us.debian.org/debian/pool/main/b/bluez/libbluetooth-dev_5.23-2+b1_i386.deb
 http://ftp.us.debian.org/debian/pool/main/b/brltty/libbrlapi0.6_5.2~20141018-5_i386.deb
 http://ftp.us.debian.org/debian/pool/main/b/brltty/libbrlapi-dev_5.2~20141018-5_i386.deb
 http://ftp.us.debian.org/debian/pool/main/c/cairo/libcairo2_1.14.0-2.1+deb8u2_i386.deb
@@ -21,6 +19,8 @@ http://ftp.us.debian.org/debian/pool/main/c/cairo/libcairo-script-interpreter2_1
 http://ftp.us.debian.org/debian/pool/main/d/dbus-glib/libdbus-glib-1-2_0.102-1_i386.deb
 http://ftp.us.debian.org/debian/pool/main/d/dbus/libdbus-1-3_1.8.22-0+deb8u1_i386.deb
 http://ftp.us.debian.org/debian/pool/main/d/dbus/libdbus-1-dev_1.8.22-0+deb8u1_i386.deb
+http://ftp.us.debian.org/debian/pool/main/d/d-conf/libdconf1_0.22.0-1_i386.deb
+http://ftp.us.debian.org/debian/pool/main/d/d-conf/libdconf-dev_0.22.0-1_i386.deb
 http://ftp.us.debian.org/debian/pool/main/e/e2fsprogs/comerr-dev_2.1-1.42.12-2+b1_i386.deb
 http://ftp.us.debian.org/debian/pool/main/e/e2fsprogs/libcomerr2_1.42.12-2+b1_i386.deb
 http://ftp.us.debian.org/debian/pool/main/e/elfutils/libelf1_0.159-4.2_i386.deb
@@ -167,6 +167,8 @@ http://ftp.us.debian.org/debian/pool/main/x/x11proto-xext/x11proto-xext-dev_7.3.
 http://ftp.us.debian.org/debian/pool/main/x/x11proto-xinerama/x11proto-xinerama-dev_1.2.1-2_all.deb
 http://ftp.us.debian.org/debian/pool/main/z/zlib/zlib1g_1.2.8.dfsg-2+b1_i386.deb
 http://ftp.us.debian.org/debian/pool/main/z/zlib/zlib1g-dev_1.2.8.dfsg-2+b1_i386.deb
+http://security.debian.org/pool/updates/main/b/bluez/libbluetooth3_5.23-2+deb8u1_i386.deb
+http://security.debian.org/pool/updates/main/b/bluez/libbluetooth-dev_5.23-2+deb8u1_i386.deb
 http://security.debian.org/pool/updates/main/c/cups/libcups2_1.7.5-11+deb8u1_i386.deb
 http://security.debian.org/pool/updates/main/c/cups/libcups2-dev_1.7.5-11+deb8u1_i386.deb
 http://security.debian.org/pool/updates/main/e/expat/libexpat1_2.1.0-6+deb8u4_i386.deb
@@ -175,8 +177,8 @@ http://security.debian.org/pool/updates/main/f/fontconfig/libfontconfig1_2.11.0-
 http://security.debian.org/pool/updates/main/f/fontconfig/libfontconfig1-dev_2.11.0-6.3+deb8u1_i386.deb
 http://security.debian.org/pool/updates/main/f/freetype/libfreetype6_2.5.2-3+deb8u2_i386.deb
 http://security.debian.org/pool/updates/main/f/freetype/libfreetype6-dev_2.5.2-3+deb8u2_i386.deb
-http://security.debian.org/pool/updates/main/g/gdk-pixbuf/libgdk-pixbuf2.0-0_2.31.1-2+deb8u5_i386.deb
-http://security.debian.org/pool/updates/main/g/gdk-pixbuf/libgdk-pixbuf2.0-dev_2.31.1-2+deb8u5_i386.deb
+http://security.debian.org/pool/updates/main/g/gdk-pixbuf/libgdk-pixbuf2.0-0_2.31.1-2+deb8u6_i386.deb
+http://security.debian.org/pool/updates/main/g/gdk-pixbuf/libgdk-pixbuf2.0-dev_2.31.1-2+deb8u6_i386.deb
 http://security.debian.org/pool/updates/main/g/glibc/libc6_2.19-18+deb8u10_i386.deb
 http://security.debian.org/pool/updates/main/g/glibc/libc6-dev_2.19-18+deb8u10_i386.deb
 http://security.debian.org/pool/updates/main/g/gnutls28/libgnutls28-dev_3.3.8-6+deb8u6_i386.deb
diff --git a/build/linux/sysroot_scripts/packagelist.jessie.mipsel b/build/linux/sysroot_scripts/packagelist.jessie.mipsel
index 2e977c5d2014..4392e12ae583 100644
--- a/build/linux/sysroot_scripts/packagelist.jessie.mipsel
+++ b/build/linux/sysroot_scripts/packagelist.jessie.mipsel
@@ -10,8 +10,6 @@ http://ftp.us.debian.org/debian/pool/main/a/attr/libattr1_2.4.47-2_mipsel.deb
 http://ftp.us.debian.org/debian/pool/main/a/audit/libaudit1_2.4-1+b1_mipsel.deb
 http://ftp.us.debian.org/debian/pool/main/a/avahi/libavahi-client3_0.6.31-5_mipsel.deb
 http://ftp.us.debian.org/debian/pool/main/a/avahi/libavahi-common3_0.6.31-5_mipsel.deb
-http://ftp.us.debian.org/debian/pool/main/b/bluez/libbluetooth3_5.23-2+b1_mipsel.deb
-http://ftp.us.debian.org/debian/pool/main/b/bluez/libbluetooth-dev_5.23-2+b1_mipsel.deb
 http://ftp.us.debian.org/debian/pool/main/b/brltty/libbrlapi0.6_5.2~20141018-5_mipsel.deb
 http://ftp.us.debian.org/debian/pool/main/b/brltty/libbrlapi-dev_5.2~20141018-5_mipsel.deb
 http://ftp.us.debian.org/debian/pool/main/c/cairo/libcairo2_1.14.0-2.1+deb8u2_mipsel.deb
@@ -21,6 +19,8 @@ http://ftp.us.debian.org/debian/pool/main/c/cairo/libcairo-script-interpreter2_1
 http://ftp.us.debian.org/debian/pool/main/d/dbus-glib/libdbus-glib-1-2_0.102-1_mipsel.deb
 http://ftp.us.debian.org/debian/pool/main/d/dbus/libdbus-1-3_1.8.22-0+deb8u1_mipsel.deb
 http://ftp.us.debian.org/debian/pool/main/d/dbus/libdbus-1-dev_1.8.22-0+deb8u1_mipsel.deb
+http://ftp.us.debian.org/debian/pool/main/d/d-conf/libdconf1_0.22.0-1_mipsel.deb
+http://ftp.us.debian.org/debian/pool/main/d/d-conf/libdconf-dev_0.22.0-1_mipsel.deb
 http://ftp.us.debian.org/debian/pool/main/e/e2fsprogs/comerr-dev_2.1-1.42.12-2+b1_mipsel.deb
 http://ftp.us.debian.org/debian/pool/main/e/e2fsprogs/libcomerr2_1.42.12-2+b1_mipsel.deb
 http://ftp.us.debian.org/debian/pool/main/e/elfutils/libelf1_0.159-4.2_mipsel.deb
@@ -161,6 +161,8 @@ http://ftp.us.debian.org/debian/pool/main/x/x11proto-xext/x11proto-xext-dev_7.3.
 http://ftp.us.debian.org/debian/pool/main/x/x11proto-xinerama/x11proto-xinerama-dev_1.2.1-2_all.deb
 http://ftp.us.debian.org/debian/pool/main/z/zlib/zlib1g_1.2.8.dfsg-2+b1_mipsel.deb
 http://ftp.us.debian.org/debian/pool/main/z/zlib/zlib1g-dev_1.2.8.dfsg-2+b1_mipsel.deb
+http://security.debian.org/pool/updates/main/b/bluez/libbluetooth3_5.23-2+deb8u1_mipsel.deb
+http://security.debian.org/pool/updates/main/b/bluez/libbluetooth-dev_5.23-2+deb8u1_mipsel.deb
 http://security.debian.org/pool/updates/main/c/cups/libcups2_1.7.5-11+deb8u1_mipsel.deb
 http://security.debian.org/pool/updates/main/c/cups/libcups2-dev_1.7.5-11+deb8u1_mipsel.deb
 http://security.debian.org/pool/updates/main/e/expat/libexpat1_2.1.0-6+deb8u4_mipsel.deb
@@ -169,8 +171,8 @@ http://security.debian.org/pool/updates/main/f/fontconfig/libfontconfig1_2.11.0-
 http://security.debian.org/pool/updates/main/f/fontconfig/libfontconfig1-dev_2.11.0-6.3+deb8u1_mipsel.deb
 http://security.debian.org/pool/updates/main/f/freetype/libfreetype6_2.5.2-3+deb8u2_mipsel.deb
 http://security.debian.org/pool/updates/main/f/freetype/libfreetype6-dev_2.5.2-3+deb8u2_mipsel.deb
-http://security.debian.org/pool/updates/main/g/gdk-pixbuf/libgdk-pixbuf2.0-0_2.31.1-2+deb8u5_mipsel.deb
-http://security.debian.org/pool/updates/main/g/gdk-pixbuf/libgdk-pixbuf2.0-dev_2.31.1-2+deb8u5_mipsel.deb
+http://security.debian.org/pool/updates/main/g/gdk-pixbuf/libgdk-pixbuf2.0-0_2.31.1-2+deb8u6_mipsel.deb
+http://security.debian.org/pool/updates/main/g/gdk-pixbuf/libgdk-pixbuf2.0-dev_2.31.1-2+deb8u6_mipsel.deb
 http://security.debian.org/pool/updates/main/g/glibc/libc6_2.19-18+deb8u10_mipsel.deb
 http://security.debian.org/pool/updates/main/g/glibc/libc6-dev_2.19-18+deb8u10_mipsel.deb
 http://security.debian.org/pool/updates/main/g/gnutls28/libgnutls28-dev_3.3.8-6+deb8u6_mipsel.deb
diff --git a/build/linux/sysroot_scripts/packagelist.stretch.amd64 b/build/linux/sysroot_scripts/packagelist.stretch.amd64
index 3bbb3da74c59..39f8b9e098c4 100644
--- a/build/linux/sysroot_scripts/packagelist.stretch.amd64
+++ b/build/linux/sysroot_scripts/packagelist.stretch.amd64
@@ -10,8 +10,6 @@ http://ftp.us.debian.org/debian/pool/main/a/attr/libattr1_2.4.47-2+b2_amd64.deb
 http://ftp.us.debian.org/debian/pool/main/a/audit/libaudit1_2.6.7-2_amd64.deb
 http://ftp.us.debian.org/debian/pool/main/a/avahi/libavahi-client3_0.6.32-2_amd64.deb
 http://ftp.us.debian.org/debian/pool/main/a/avahi/libavahi-common3_0.6.32-2_amd64.deb
-http://ftp.us.debian.org/debian/pool/main/b/bluez/libbluetooth3_5.43-2_amd64.deb
-http://ftp.us.debian.org/debian/pool/main/b/bluez/libbluetooth-dev_5.43-2_amd64.deb
 http://ftp.us.debian.org/debian/pool/main/b/brltty/libbrlapi0.6_5.4-7_amd64.deb
 http://ftp.us.debian.org/debian/pool/main/b/brltty/libbrlapi-dev_5.4-7_amd64.deb
 http://ftp.us.debian.org/debian/pool/main/c/cairo/libcairo2_1.14.8-1_amd64.deb
@@ -25,6 +23,8 @@ http://ftp.us.debian.org/debian/pool/main/c/cups/libcupsimage2-dev_2.2.1-8_amd64
 http://ftp.us.debian.org/debian/pool/main/d/dbus-glib/libdbus-glib-1-2_0.108-2_amd64.deb
 http://ftp.us.debian.org/debian/pool/main/d/dbus/libdbus-1-3_1.10.18-1_amd64.deb
 http://ftp.us.debian.org/debian/pool/main/d/dbus/libdbus-1-dev_1.10.18-1_amd64.deb
+http://ftp.us.debian.org/debian/pool/main/d/d-conf/libdconf1_0.26.0-2+b1_amd64.deb
+http://ftp.us.debian.org/debian/pool/main/d/d-conf/libdconf-dev_0.26.0-2+b1_amd64.deb
 http://ftp.us.debian.org/debian/pool/main/e/e2fsprogs/comerr-dev_2.1-1.43.4-2_amd64.deb
 http://ftp.us.debian.org/debian/pool/main/e/e2fsprogs/libcomerr2_1.43.4-2_amd64.deb
 http://ftp.us.debian.org/debian/pool/main/e/elfutils/libelf1_0.168-1_amd64.deb
@@ -50,8 +50,6 @@ http://ftp.us.debian.org/debian/pool/main/g/gcc-6/libubsan0_6.3.0-18_amd64.deb
 http://ftp.us.debian.org/debian/pool/main/g/gconf/libgconf2-4_3.2.6-4+b1_amd64.deb
 http://ftp.us.debian.org/debian/pool/main/g/gconf/libgconf-2-4_3.2.6-4+b1_amd64.deb
 http://ftp.us.debian.org/debian/pool/main/g/gconf/libgconf2-dev_3.2.6-4+b1_amd64.deb
-http://ftp.us.debian.org/debian/pool/main/g/gdk-pixbuf/libgdk-pixbuf2.0-0_2.36.5-2_amd64.deb
-http://ftp.us.debian.org/debian/pool/main/g/gdk-pixbuf/libgdk-pixbuf2.0-dev_2.36.5-2_amd64.deb
 http://ftp.us.debian.org/debian/pool/main/g/glib2.0/libglib2.0-0_2.50.3-2_amd64.deb
 http://ftp.us.debian.org/debian/pool/main/g/glib2.0/libglib2.0-dev_2.50.3-2_amd64.deb
 http://ftp.us.debian.org/debian/pool/main/g/graphite2/libgraphite2-3_1.3.10-1_amd64.deb
@@ -209,8 +207,12 @@ http://ftp.us.debian.org/debian/pool/main/x/x11proto-xext/x11proto-xext-dev_7.3.
 http://ftp.us.debian.org/debian/pool/main/x/x11proto-xinerama/x11proto-xinerama-dev_1.2.1-2_all.deb
 http://ftp.us.debian.org/debian/pool/main/z/zlib/zlib1g_1.2.8.dfsg-5_amd64.deb
 http://ftp.us.debian.org/debian/pool/main/z/zlib/zlib1g-dev_1.2.8.dfsg-5_amd64.deb
+http://security.debian.org/pool/updates/main/b/bluez/libbluetooth3_5.43-2+deb9u1_amd64.deb
+http://security.debian.org/pool/updates/main/b/bluez/libbluetooth-dev_5.43-2+deb9u1_amd64.deb
 http://security.debian.org/pool/updates/main/e/expat/libexpat1_2.2.0-2+deb9u1_amd64.deb
 http://security.debian.org/pool/updates/main/e/expat/libexpat1-dev_2.2.0-2+deb9u1_amd64.deb
+http://security.debian.org/pool/updates/main/g/gdk-pixbuf/libgdk-pixbuf2.0-0_2.36.5-2+deb9u1_amd64.deb
+http://security.debian.org/pool/updates/main/g/gdk-pixbuf/libgdk-pixbuf2.0-dev_2.36.5-2+deb9u1_amd64.deb
 http://security.debian.org/pool/updates/main/g/glibc/libc6_2.24-11+deb9u1_amd64.deb
 http://security.debian.org/pool/updates/main/g/glibc/libc6-dev_2.24-11+deb9u1_amd64.deb
 http://security.debian.org/pool/updates/main/g/gnutls28/libgnutls28-dev_3.5.8-5+deb9u1_amd64.deb
@@ -218,6 +220,6 @@ http://security.debian.org/pool/updates/main/g/gnutls28/libgnutls30_3.5.8-5+deb9
 http://security.debian.org/pool/updates/main/g/gnutls28/libgnutls-dane0_3.5.8-5+deb9u1_amd64.deb
 http://security.debian.org/pool/updates/main/g/gnutls28/libgnutls-openssl27_3.5.8-5+deb9u1_amd64.deb
 http://security.debian.org/pool/updates/main/g/gnutls28/libgnutlsxx28_3.5.8-5+deb9u1_amd64.deb
-http://security.debian.org/pool/updates/main/libg/libgcrypt20/libgcrypt20_1.7.6-2+deb9u1_amd64.deb
-http://security.debian.org/pool/updates/main/libg/libgcrypt20/libgcrypt20-dev_1.7.6-2+deb9u1_amd64.deb
-http://security.debian.org/pool/updates/main/l/linux/linux-libc-dev_4.9.30-2+deb9u2_amd64.deb
+http://security.debian.org/pool/updates/main/libg/libgcrypt20/libgcrypt20_1.7.6-2+deb9u2_amd64.deb
+http://security.debian.org/pool/updates/main/libg/libgcrypt20/libgcrypt20-dev_1.7.6-2+deb9u2_amd64.deb
+http://security.debian.org/pool/updates/main/l/linux/linux-libc-dev_4.9.30-2+deb9u5_amd64.deb
diff --git a/build/linux/sysroot_scripts/packagelist.stretch.arm b/build/linux/sysroot_scripts/packagelist.stretch.arm
index b57c37e8b8fd..e87baa316e7e 100644
--- a/build/linux/sysroot_scripts/packagelist.stretch.arm
+++ b/build/linux/sysroot_scripts/packagelist.stretch.arm
@@ -10,8 +10,6 @@ http://ftp.us.debian.org/debian/pool/main/a/attr/libattr1_2.4.47-2+b2_armhf.deb
 http://ftp.us.debian.org/debian/pool/main/a/audit/libaudit1_2.6.7-2_armhf.deb
 http://ftp.us.debian.org/debian/pool/main/a/avahi/libavahi-client3_0.6.32-2_armhf.deb
 http://ftp.us.debian.org/debian/pool/main/a/avahi/libavahi-common3_0.6.32-2_armhf.deb
-http://ftp.us.debian.org/debian/pool/main/b/bluez/libbluetooth3_5.43-2_armhf.deb
-http://ftp.us.debian.org/debian/pool/main/b/bluez/libbluetooth-dev_5.43-2_armhf.deb
 http://ftp.us.debian.org/debian/pool/main/b/brltty/libbrlapi0.6_5.4-7_armhf.deb
 http://ftp.us.debian.org/debian/pool/main/b/brltty/libbrlapi-dev_5.4-7_armhf.deb
 http://ftp.us.debian.org/debian/pool/main/c/cairo/libcairo2_1.14.8-1_armhf.deb
@@ -25,6 +23,8 @@ http://ftp.us.debian.org/debian/pool/main/c/cups/libcupsimage2-dev_2.2.1-8_armhf
 http://ftp.us.debian.org/debian/pool/main/d/dbus-glib/libdbus-glib-1-2_0.108-2_armhf.deb
 http://ftp.us.debian.org/debian/pool/main/d/dbus/libdbus-1-3_1.10.18-1_armhf.deb
 http://ftp.us.debian.org/debian/pool/main/d/dbus/libdbus-1-dev_1.10.18-1_armhf.deb
+http://ftp.us.debian.org/debian/pool/main/d/d-conf/libdconf1_0.26.0-2+b1_armhf.deb
+http://ftp.us.debian.org/debian/pool/main/d/d-conf/libdconf-dev_0.26.0-2+b1_armhf.deb
 http://ftp.us.debian.org/debian/pool/main/e/e2fsprogs/comerr-dev_2.1-1.43.4-2_armhf.deb
 http://ftp.us.debian.org/debian/pool/main/e/e2fsprogs/libcomerr2_1.43.4-2_armhf.deb
 http://ftp.us.debian.org/debian/pool/main/e/elfutils/libelf1_0.168-1_armhf.deb
@@ -44,8 +44,6 @@ http://ftp.us.debian.org/debian/pool/main/g/gcc-6/libubsan0_6.3.0-18_armhf.deb
 http://ftp.us.debian.org/debian/pool/main/g/gconf/libgconf2-4_3.2.6-4+b1_armhf.deb
 http://ftp.us.debian.org/debian/pool/main/g/gconf/libgconf-2-4_3.2.6-4+b1_armhf.deb
 http://ftp.us.debian.org/debian/pool/main/g/gconf/libgconf2-dev_3.2.6-4+b1_armhf.deb
-http://ftp.us.debian.org/debian/pool/main/g/gdk-pixbuf/libgdk-pixbuf2.0-0_2.36.5-2_armhf.deb
-http://ftp.us.debian.org/debian/pool/main/g/gdk-pixbuf/libgdk-pixbuf2.0-dev_2.36.5-2_armhf.deb
 http://ftp.us.debian.org/debian/pool/main/g/glib2.0/libglib2.0-0_2.50.3-2_armhf.deb
 http://ftp.us.debian.org/debian/pool/main/g/glib2.0/libglib2.0-dev_2.50.3-2_armhf.deb
 http://ftp.us.debian.org/debian/pool/main/g/graphite2/libgraphite2-3_1.3.10-1_armhf.deb
@@ -206,8 +204,12 @@ http://ftp.us.debian.org/debian/pool/main/x/x11proto-xext/x11proto-xext-dev_7.3.
 http://ftp.us.debian.org/debian/pool/main/x/x11proto-xinerama/x11proto-xinerama-dev_1.2.1-2_all.deb
 http://ftp.us.debian.org/debian/pool/main/z/zlib/zlib1g_1.2.8.dfsg-5_armhf.deb
 http://ftp.us.debian.org/debian/pool/main/z/zlib/zlib1g-dev_1.2.8.dfsg-5_armhf.deb
+http://security.debian.org/pool/updates/main/b/bluez/libbluetooth3_5.43-2+deb9u1_armhf.deb
+http://security.debian.org/pool/updates/main/b/bluez/libbluetooth-dev_5.43-2+deb9u1_armhf.deb
 http://security.debian.org/pool/updates/main/e/expat/libexpat1_2.2.0-2+deb9u1_armhf.deb
 http://security.debian.org/pool/updates/main/e/expat/libexpat1-dev_2.2.0-2+deb9u1_armhf.deb
+http://security.debian.org/pool/updates/main/g/gdk-pixbuf/libgdk-pixbuf2.0-0_2.36.5-2+deb9u1_armhf.deb
+http://security.debian.org/pool/updates/main/g/gdk-pixbuf/libgdk-pixbuf2.0-dev_2.36.5-2+deb9u1_armhf.deb
 http://security.debian.org/pool/updates/main/g/glibc/libc6_2.24-11+deb9u1_armhf.deb
 http://security.debian.org/pool/updates/main/g/glibc/libc6-dev_2.24-11+deb9u1_armhf.deb
 http://security.debian.org/pool/updates/main/g/gnutls28/libgnutls28-dev_3.5.8-5+deb9u1_armhf.deb
@@ -215,6 +217,6 @@ http://security.debian.org/pool/updates/main/g/gnutls28/libgnutls30_3.5.8-5+deb9
 http://security.debian.org/pool/updates/main/g/gnutls28/libgnutls-dane0_3.5.8-5+deb9u1_armhf.deb
 http://security.debian.org/pool/updates/main/g/gnutls28/libgnutls-openssl27_3.5.8-5+deb9u1_armhf.deb
 http://security.debian.org/pool/updates/main/g/gnutls28/libgnutlsxx28_3.5.8-5+deb9u1_armhf.deb
-http://security.debian.org/pool/updates/main/libg/libgcrypt20/libgcrypt20_1.7.6-2+deb9u1_armhf.deb
-http://security.debian.org/pool/updates/main/libg/libgcrypt20/libgcrypt20-dev_1.7.6-2+deb9u1_armhf.deb
-http://security.debian.org/pool/updates/main/l/linux/linux-libc-dev_4.9.30-2+deb9u2_armhf.deb
+http://security.debian.org/pool/updates/main/libg/libgcrypt20/libgcrypt20_1.7.6-2+deb9u2_armhf.deb
+http://security.debian.org/pool/updates/main/libg/libgcrypt20/libgcrypt20-dev_1.7.6-2+deb9u2_armhf.deb
+http://security.debian.org/pool/updates/main/l/linux/linux-libc-dev_4.9.30-2+deb9u5_armhf.deb
diff --git a/build/linux/sysroot_scripts/packagelist.stretch.arm64 b/build/linux/sysroot_scripts/packagelist.stretch.arm64
index 2e79d8377e45..cd75e2cdefc1 100644
--- a/build/linux/sysroot_scripts/packagelist.stretch.arm64
+++ b/build/linux/sysroot_scripts/packagelist.stretch.arm64
@@ -10,8 +10,6 @@ http://ftp.us.debian.org/debian/pool/main/a/attr/libattr1_2.4.47-2+b2_arm64.deb
 http://ftp.us.debian.org/debian/pool/main/a/audit/libaudit1_2.6.7-2_arm64.deb
 http://ftp.us.debian.org/debian/pool/main/a/avahi/libavahi-client3_0.6.32-2_arm64.deb
 http://ftp.us.debian.org/debian/pool/main/a/avahi/libavahi-common3_0.6.32-2_arm64.deb
-http://ftp.us.debian.org/debian/pool/main/b/bluez/libbluetooth3_5.43-2_arm64.deb
-http://ftp.us.debian.org/debian/pool/main/b/bluez/libbluetooth-dev_5.43-2_arm64.deb
 http://ftp.us.debian.org/debian/pool/main/b/brltty/libbrlapi0.6_5.4-7_arm64.deb
 http://ftp.us.debian.org/debian/pool/main/b/brltty/libbrlapi-dev_5.4-7_arm64.deb
 http://ftp.us.debian.org/debian/pool/main/c/cairo/libcairo2_1.14.8-1_arm64.deb
@@ -25,6 +23,8 @@ http://ftp.us.debian.org/debian/pool/main/c/cups/libcupsimage2-dev_2.2.1-8_arm64
 http://ftp.us.debian.org/debian/pool/main/d/dbus-glib/libdbus-glib-1-2_0.108-2_arm64.deb
 http://ftp.us.debian.org/debian/pool/main/d/dbus/libdbus-1-3_1.10.18-1_arm64.deb
 http://ftp.us.debian.org/debian/pool/main/d/dbus/libdbus-1-dev_1.10.18-1_arm64.deb
+http://ftp.us.debian.org/debian/pool/main/d/d-conf/libdconf1_0.26.0-2+b1_arm64.deb
+http://ftp.us.debian.org/debian/pool/main/d/d-conf/libdconf-dev_0.26.0-2+b1_arm64.deb
 http://ftp.us.debian.org/debian/pool/main/e/e2fsprogs/comerr-dev_2.1-1.43.4-2_arm64.deb
 http://ftp.us.debian.org/debian/pool/main/e/e2fsprogs/libcomerr2_1.43.4-2_arm64.deb
 http://ftp.us.debian.org/debian/pool/main/e/elfutils/libelf1_0.168-1_arm64.deb
@@ -45,8 +45,6 @@ http://ftp.us.debian.org/debian/pool/main/g/gcc-6/libubsan0_6.3.0-18_arm64.deb
 http://ftp.us.debian.org/debian/pool/main/g/gconf/libgconf2-4_3.2.6-4+b1_arm64.deb
 http://ftp.us.debian.org/debian/pool/main/g/gconf/libgconf-2-4_3.2.6-4+b1_arm64.deb
 http://ftp.us.debian.org/debian/pool/main/g/gconf/libgconf2-dev_3.2.6-4+b1_arm64.deb
-http://ftp.us.debian.org/debian/pool/main/g/gdk-pixbuf/libgdk-pixbuf2.0-0_2.36.5-2_arm64.deb
-http://ftp.us.debian.org/debian/pool/main/g/gdk-pixbuf/libgdk-pixbuf2.0-dev_2.36.5-2_arm64.deb
 http://ftp.us.debian.org/debian/pool/main/g/glib2.0/libglib2.0-0_2.50.3-2_arm64.deb
 http://ftp.us.debian.org/debian/pool/main/g/glib2.0/libglib2.0-dev_2.50.3-2_arm64.deb
 http://ftp.us.debian.org/debian/pool/main/g/gmp/libgmp10_6.1.2+dfsg-1_arm64.deb
@@ -208,8 +206,12 @@ http://ftp.us.debian.org/debian/pool/main/x/x11proto-xext/x11proto-xext-dev_7.3.
 http://ftp.us.debian.org/debian/pool/main/x/x11proto-xinerama/x11proto-xinerama-dev_1.2.1-2_all.deb
 http://ftp.us.debian.org/debian/pool/main/z/zlib/zlib1g_1.2.8.dfsg-5_arm64.deb
 http://ftp.us.debian.org/debian/pool/main/z/zlib/zlib1g-dev_1.2.8.dfsg-5_arm64.deb
+http://security.debian.org/pool/updates/main/b/bluez/libbluetooth3_5.43-2+deb9u1_arm64.deb
+http://security.debian.org/pool/updates/main/b/bluez/libbluetooth-dev_5.43-2+deb9u1_arm64.deb
 http://security.debian.org/pool/updates/main/e/expat/libexpat1_2.2.0-2+deb9u1_arm64.deb
 http://security.debian.org/pool/updates/main/e/expat/libexpat1-dev_2.2.0-2+deb9u1_arm64.deb
+http://security.debian.org/pool/updates/main/g/gdk-pixbuf/libgdk-pixbuf2.0-0_2.36.5-2+deb9u1_arm64.deb
+http://security.debian.org/pool/updates/main/g/gdk-pixbuf/libgdk-pixbuf2.0-dev_2.36.5-2+deb9u1_arm64.deb
 http://security.debian.org/pool/updates/main/g/glibc/libc6_2.24-11+deb9u1_arm64.deb
 http://security.debian.org/pool/updates/main/g/glibc/libc6-dev_2.24-11+deb9u1_arm64.deb
 http://security.debian.org/pool/updates/main/g/gnutls28/libgnutls28-dev_3.5.8-5+deb9u1_arm64.deb
@@ -217,6 +219,6 @@ http://security.debian.org/pool/updates/main/g/gnutls28/libgnutls30_3.5.8-5+deb9
 http://security.debian.org/pool/updates/main/g/gnutls28/libgnutls-dane0_3.5.8-5+deb9u1_arm64.deb
 http://security.debian.org/pool/updates/main/g/gnutls28/libgnutls-openssl27_3.5.8-5+deb9u1_arm64.deb
 http://security.debian.org/pool/updates/main/g/gnutls28/libgnutlsxx28_3.5.8-5+deb9u1_arm64.deb
-http://security.debian.org/pool/updates/main/libg/libgcrypt20/libgcrypt20_1.7.6-2+deb9u1_arm64.deb
-http://security.debian.org/pool/updates/main/libg/libgcrypt20/libgcrypt20-dev_1.7.6-2+deb9u1_arm64.deb
-http://security.debian.org/pool/updates/main/l/linux/linux-libc-dev_4.9.30-2+deb9u2_arm64.deb
+http://security.debian.org/pool/updates/main/libg/libgcrypt20/libgcrypt20_1.7.6-2+deb9u2_arm64.deb
+http://security.debian.org/pool/updates/main/libg/libgcrypt20/libgcrypt20-dev_1.7.6-2+deb9u2_arm64.deb
+http://security.debian.org/pool/updates/main/l/linux/linux-libc-dev_4.9.30-2+deb9u5_arm64.deb
diff --git a/build/linux/sysroot_scripts/packagelist.stretch.i386 b/build/linux/sysroot_scripts/packagelist.stretch.i386
index 496f4c4c0536..6c803a1e8300 100644
--- a/build/linux/sysroot_scripts/packagelist.stretch.i386
+++ b/build/linux/sysroot_scripts/packagelist.stretch.i386
@@ -10,8 +10,6 @@ http://ftp.us.debian.org/debian/pool/main/a/attr/libattr1_2.4.47-2+b2_i386.deb
 http://ftp.us.debian.org/debian/pool/main/a/audit/libaudit1_2.6.7-2_i386.deb
 http://ftp.us.debian.org/debian/pool/main/a/avahi/libavahi-client3_0.6.32-2_i386.deb
 http://ftp.us.debian.org/debian/pool/main/a/avahi/libavahi-common3_0.6.32-2_i386.deb
-http://ftp.us.debian.org/debian/pool/main/b/bluez/libbluetooth3_5.43-2_i386.deb
-http://ftp.us.debian.org/debian/pool/main/b/bluez/libbluetooth-dev_5.43-2_i386.deb
 http://ftp.us.debian.org/debian/pool/main/b/brltty/libbrlapi0.6_5.4-7_i386.deb
 http://ftp.us.debian.org/debian/pool/main/b/brltty/libbrlapi-dev_5.4-7_i386.deb
 http://ftp.us.debian.org/debian/pool/main/c/cairo/libcairo2_1.14.8-1_i386.deb
@@ -25,6 +23,8 @@ http://ftp.us.debian.org/debian/pool/main/c/cups/libcupsimage2-dev_2.2.1-8_i386.
 http://ftp.us.debian.org/debian/pool/main/d/dbus-glib/libdbus-glib-1-2_0.108-2_i386.deb
 http://ftp.us.debian.org/debian/pool/main/d/dbus/libdbus-1-3_1.10.18-1_i386.deb
 http://ftp.us.debian.org/debian/pool/main/d/dbus/libdbus-1-dev_1.10.18-1_i386.deb
+http://ftp.us.debian.org/debian/pool/main/d/d-conf/libdconf1_0.26.0-2+b1_i386.deb
+http://ftp.us.debian.org/debian/pool/main/d/d-conf/libdconf-dev_0.26.0-2+b1_i386.deb
 http://ftp.us.debian.org/debian/pool/main/e/e2fsprogs/comerr-dev_2.1-1.43.4-2_i386.deb
 http://ftp.us.debian.org/debian/pool/main/e/e2fsprogs/libcomerr2_1.43.4-2_i386.deb
 http://ftp.us.debian.org/debian/pool/main/e/elfutils/libelf1_0.168-1_i386.deb
@@ -48,8 +48,6 @@ http://ftp.us.debian.org/debian/pool/main/g/gcc-6/libubsan0_6.3.0-18_i386.deb
 http://ftp.us.debian.org/debian/pool/main/g/gconf/libgconf-2-4_3.2.6-4+b1_i386.deb
 http://ftp.us.debian.org/debian/pool/main/g/gconf/libgconf2-4_3.2.6-4+b1_i386.deb
 http://ftp.us.debian.org/debian/pool/main/g/gconf/libgconf2-dev_3.2.6-4+b1_i386.deb
-http://ftp.us.debian.org/debian/pool/main/g/gdk-pixbuf/libgdk-pixbuf2.0-0_2.36.5-2_i386.deb
-http://ftp.us.debian.org/debian/pool/main/g/gdk-pixbuf/libgdk-pixbuf2.0-dev_2.36.5-2_i386.deb
 http://ftp.us.debian.org/debian/pool/main/g/glib2.0/libglib2.0-0_2.50.3-2_i386.deb
 http://ftp.us.debian.org/debian/pool/main/g/glib2.0/libglib2.0-dev_2.50.3-2_i386.deb
 http://ftp.us.debian.org/debian/pool/main/g/graphite2/libgraphite2-3_1.3.10-1_i386.deb
@@ -207,8 +205,12 @@ http://ftp.us.debian.org/debian/pool/main/x/x11proto-xext/x11proto-xext-dev_7.3.
 http://ftp.us.debian.org/debian/pool/main/x/x11proto-xinerama/x11proto-xinerama-dev_1.2.1-2_all.deb
 http://ftp.us.debian.org/debian/pool/main/z/zlib/zlib1g_1.2.8.dfsg-5_i386.deb
 http://ftp.us.debian.org/debian/pool/main/z/zlib/zlib1g-dev_1.2.8.dfsg-5_i386.deb
+http://security.debian.org/pool/updates/main/b/bluez/libbluetooth3_5.43-2+deb9u1_i386.deb
+http://security.debian.org/pool/updates/main/b/bluez/libbluetooth-dev_5.43-2+deb9u1_i386.deb
 http://security.debian.org/pool/updates/main/e/expat/libexpat1_2.2.0-2+deb9u1_i386.deb
 http://security.debian.org/pool/updates/main/e/expat/libexpat1-dev_2.2.0-2+deb9u1_i386.deb
+http://security.debian.org/pool/updates/main/g/gdk-pixbuf/libgdk-pixbuf2.0-0_2.36.5-2+deb9u1_i386.deb
+http://security.debian.org/pool/updates/main/g/gdk-pixbuf/libgdk-pixbuf2.0-dev_2.36.5-2+deb9u1_i386.deb
 http://security.debian.org/pool/updates/main/g/glibc/libc6_2.24-11+deb9u1_i386.deb
 http://security.debian.org/pool/updates/main/g/glibc/libc6-dev_2.24-11+deb9u1_i386.deb
 http://security.debian.org/pool/updates/main/g/gnutls28/libgnutls28-dev_3.5.8-5+deb9u1_i386.deb
@@ -216,6 +218,6 @@ http://security.debian.org/pool/updates/main/g/gnutls28/libgnutls30_3.5.8-5+deb9
 http://security.debian.org/pool/updates/main/g/gnutls28/libgnutls-dane0_3.5.8-5+deb9u1_i386.deb
 http://security.debian.org/pool/updates/main/g/gnutls28/libgnutls-openssl27_3.5.8-5+deb9u1_i386.deb
 http://security.debian.org/pool/updates/main/g/gnutls28/libgnutlsxx28_3.5.8-5+deb9u1_i386.deb
-http://security.debian.org/pool/updates/main/libg/libgcrypt20/libgcrypt20_1.7.6-2+deb9u1_i386.deb
-http://security.debian.org/pool/updates/main/libg/libgcrypt20/libgcrypt20-dev_1.7.6-2+deb9u1_i386.deb
-http://security.debian.org/pool/updates/main/l/linux/linux-libc-dev_4.9.30-2+deb9u2_i386.deb
+http://security.debian.org/pool/updates/main/libg/libgcrypt20/libgcrypt20_1.7.6-2+deb9u2_i386.deb
+http://security.debian.org/pool/updates/main/libg/libgcrypt20/libgcrypt20-dev_1.7.6-2+deb9u2_i386.deb
+http://security.debian.org/pool/updates/main/l/linux/linux-libc-dev_4.9.30-2+deb9u5_i386.deb
diff --git a/build/linux/sysroot_scripts/packagelist.stretch.mips64el b/build/linux/sysroot_scripts/packagelist.stretch.mips64el
index 6278a16824b9..9179d18f6eaa 100644
--- a/build/linux/sysroot_scripts/packagelist.stretch.mips64el
+++ b/build/linux/sysroot_scripts/packagelist.stretch.mips64el
@@ -10,8 +10,6 @@ http://ftp.us.debian.org/debian/pool/main/a/attr/libattr1_2.4.47-2+b2_mips64el.d
 http://ftp.us.debian.org/debian/pool/main/a/audit/libaudit1_2.6.7-2_mips64el.deb
 http://ftp.us.debian.org/debian/pool/main/a/avahi/libavahi-client3_0.6.32-2_mips64el.deb
 http://ftp.us.debian.org/debian/pool/main/a/avahi/libavahi-common3_0.6.32-2_mips64el.deb
-http://ftp.us.debian.org/debian/pool/main/b/bluez/libbluetooth3_5.43-2_mips64el.deb
-http://ftp.us.debian.org/debian/pool/main/b/bluez/libbluetooth-dev_5.43-2_mips64el.deb
 http://ftp.us.debian.org/debian/pool/main/b/brltty/libbrlapi0.6_5.4-7_mips64el.deb
 http://ftp.us.debian.org/debian/pool/main/b/brltty/libbrlapi-dev_5.4-7_mips64el.deb
 http://ftp.us.debian.org/debian/pool/main/c/cairo/libcairo2_1.14.8-1_mips64el.deb
@@ -25,6 +23,8 @@ http://ftp.us.debian.org/debian/pool/main/c/cups/libcupsimage2-dev_2.2.1-8_mips6
 http://ftp.us.debian.org/debian/pool/main/d/dbus-glib/libdbus-glib-1-2_0.108-2_mips64el.deb
 http://ftp.us.debian.org/debian/pool/main/d/dbus/libdbus-1-3_1.10.18-1_mips64el.deb
 http://ftp.us.debian.org/debian/pool/main/d/dbus/libdbus-1-dev_1.10.18-1_mips64el.deb
+http://ftp.us.debian.org/debian/pool/main/d/d-conf/libdconf1_0.26.0-2+b1_mips64el.deb
+http://ftp.us.debian.org/debian/pool/main/d/d-conf/libdconf-dev_0.26.0-2+b1_mips64el.deb
 http://ftp.us.debian.org/debian/pool/main/e/e2fsprogs/comerr-dev_2.1-1.43.4-2_mips64el.deb
 http://ftp.us.debian.org/debian/pool/main/e/e2fsprogs/libcomerr2_1.43.4-2_mips64el.deb
 http://ftp.us.debian.org/debian/pool/main/e/elfutils/libelf1_0.168-1_mips64el.deb
@@ -42,8 +42,6 @@ http://ftp.us.debian.org/debian/pool/main/g/gcc-6/libstdc++-6-dev_6.3.0-18_mips6
 http://ftp.us.debian.org/debian/pool/main/g/gconf/libgconf-2-4_3.2.6-4+b1_mips64el.deb
 http://ftp.us.debian.org/debian/pool/main/g/gconf/libgconf2-4_3.2.6-4+b1_mips64el.deb
 http://ftp.us.debian.org/debian/pool/main/g/gconf/libgconf2-dev_3.2.6-4+b1_mips64el.deb
-http://ftp.us.debian.org/debian/pool/main/g/gdk-pixbuf/libgdk-pixbuf2.0-0_2.36.5-2_mips64el.deb
-http://ftp.us.debian.org/debian/pool/main/g/gdk-pixbuf/libgdk-pixbuf2.0-dev_2.36.5-2_mips64el.deb
 http://ftp.us.debian.org/debian/pool/main/g/glib2.0/libglib2.0-0_2.50.3-2_mips64el.deb
 http://ftp.us.debian.org/debian/pool/main/g/glib2.0/libglib2.0-dev_2.50.3-2_mips64el.deb
 http://ftp.us.debian.org/debian/pool/main/g/graphite2/libgraphite2-3_1.3.10-1_mips64el.deb
@@ -200,8 +198,12 @@ http://ftp.us.debian.org/debian/pool/main/x/x11proto-xext/x11proto-xext-dev_7.3.
 http://ftp.us.debian.org/debian/pool/main/x/x11proto-xinerama/x11proto-xinerama-dev_1.2.1-2_all.deb
 http://ftp.us.debian.org/debian/pool/main/z/zlib/zlib1g_1.2.8.dfsg-5_mips64el.deb
 http://ftp.us.debian.org/debian/pool/main/z/zlib/zlib1g-dev_1.2.8.dfsg-5_mips64el.deb
+http://security.debian.org/pool/updates/main/b/bluez/libbluetooth3_5.43-2+deb9u1_mips64el.deb
+http://security.debian.org/pool/updates/main/b/bluez/libbluetooth-dev_5.43-2+deb9u1_mips64el.deb
 http://security.debian.org/pool/updates/main/e/expat/libexpat1_2.2.0-2+deb9u1_mips64el.deb
 http://security.debian.org/pool/updates/main/e/expat/libexpat1-dev_2.2.0-2+deb9u1_mips64el.deb
+http://security.debian.org/pool/updates/main/g/gdk-pixbuf/libgdk-pixbuf2.0-0_2.36.5-2+deb9u1_mips64el.deb
+http://security.debian.org/pool/updates/main/g/gdk-pixbuf/libgdk-pixbuf2.0-dev_2.36.5-2+deb9u1_mips64el.deb
 http://security.debian.org/pool/updates/main/g/glibc/libc6_2.24-11+deb9u1_mips64el.deb
 http://security.debian.org/pool/updates/main/g/glibc/libc6-dev_2.24-11+deb9u1_mips64el.deb
 http://security.debian.org/pool/updates/main/g/gnutls28/libgnutls28-dev_3.5.8-5+deb9u1_mips64el.deb
@@ -211,4 +213,4 @@ http://security.debian.org/pool/updates/main/g/gnutls28/libgnutls-openssl27_3.5.
 http://security.debian.org/pool/updates/main/g/gnutls28/libgnutlsxx28_3.5.8-5+deb9u1_mips64el.deb
 http://security.debian.org/pool/updates/main/libg/libgcrypt20/libgcrypt20_1.7.6-2+deb9u2_mips64el.deb
 http://security.debian.org/pool/updates/main/libg/libgcrypt20/libgcrypt20-dev_1.7.6-2+deb9u2_mips64el.deb
-http://security.debian.org/pool/updates/main/l/linux/linux-libc-dev_4.9.30-2+deb9u3_mips64el.deb
+http://security.debian.org/pool/updates/main/l/linux/linux-libc-dev_4.9.30-2+deb9u5_mips64el.deb
diff --git a/build/linux/sysroot_scripts/packagelist.stretch.mipsel b/build/linux/sysroot_scripts/packagelist.stretch.mipsel
index 169269f1b0a0..8465c68d1647 100644
--- a/build/linux/sysroot_scripts/packagelist.stretch.mipsel
+++ b/build/linux/sysroot_scripts/packagelist.stretch.mipsel
@@ -10,8 +10,6 @@ http://ftp.us.debian.org/debian/pool/main/a/attr/libattr1_2.4.47-2+b2_mipsel.deb
 http://ftp.us.debian.org/debian/pool/main/a/audit/libaudit1_2.6.7-2_mipsel.deb
 http://ftp.us.debian.org/debian/pool/main/a/avahi/libavahi-client3_0.6.32-2_mipsel.deb
 http://ftp.us.debian.org/debian/pool/main/a/avahi/libavahi-common3_0.6.32-2_mipsel.deb
-http://ftp.us.debian.org/debian/pool/main/b/bluez/libbluetooth3_5.43-2_mipsel.deb
-http://ftp.us.debian.org/debian/pool/main/b/bluez/libbluetooth-dev_5.43-2_mipsel.deb
 http://ftp.us.debian.org/debian/pool/main/b/brltty/libbrlapi0.6_5.4-7_mipsel.deb
 http://ftp.us.debian.org/debian/pool/main/b/brltty/libbrlapi-dev_5.4-7_mipsel.deb
 http://ftp.us.debian.org/debian/pool/main/c/cairo/libcairo2_1.14.8-1_mipsel.deb
@@ -25,6 +23,8 @@ http://ftp.us.debian.org/debian/pool/main/c/cups/libcupsimage2-dev_2.2.1-8_mipse
 http://ftp.us.debian.org/debian/pool/main/d/dbus-glib/libdbus-glib-1-2_0.108-2_mipsel.deb
 http://ftp.us.debian.org/debian/pool/main/d/dbus/libdbus-1-3_1.10.18-1_mipsel.deb
 http://ftp.us.debian.org/debian/pool/main/d/dbus/libdbus-1-dev_1.10.18-1_mipsel.deb
+http://ftp.us.debian.org/debian/pool/main/d/d-conf/libdconf1_0.26.0-2+b1_mipsel.deb
+http://ftp.us.debian.org/debian/pool/main/d/d-conf/libdconf-dev_0.26.0-2+b1_mipsel.deb
 http://ftp.us.debian.org/debian/pool/main/e/e2fsprogs/comerr-dev_2.1-1.43.4-2_mipsel.deb
 http://ftp.us.debian.org/debian/pool/main/e/e2fsprogs/libcomerr2_1.43.4-2_mipsel.deb
 http://ftp.us.debian.org/debian/pool/main/e/elfutils/libelf1_0.168-1_mipsel.deb
@@ -42,8 +42,6 @@ http://ftp.us.debian.org/debian/pool/main/g/gcc-6/libstdc++-6-dev_6.3.0-18_mipse
 http://ftp.us.debian.org/debian/pool/main/g/gconf/libgconf2-4_3.2.6-4+b1_mipsel.deb
 http://ftp.us.debian.org/debian/pool/main/g/gconf/libgconf-2-4_3.2.6-4+b1_mipsel.deb
 http://ftp.us.debian.org/debian/pool/main/g/gconf/libgconf2-dev_3.2.6-4+b1_mipsel.deb
-http://ftp.us.debian.org/debian/pool/main/g/gdk-pixbuf/libgdk-pixbuf2.0-0_2.36.5-2_mipsel.deb
-http://ftp.us.debian.org/debian/pool/main/g/gdk-pixbuf/libgdk-pixbuf2.0-dev_2.36.5-2_mipsel.deb
 http://ftp.us.debian.org/debian/pool/main/g/glib2.0/libglib2.0-0_2.50.3-2_mipsel.deb
 http://ftp.us.debian.org/debian/pool/main/g/glib2.0/libglib2.0-dev_2.50.3-2_mipsel.deb
 http://ftp.us.debian.org/debian/pool/main/g/graphite2/libgraphite2-3_1.3.10-1_mipsel.deb
@@ -200,8 +198,12 @@ http://ftp.us.debian.org/debian/pool/main/x/x11proto-xext/x11proto-xext-dev_7.3.
 http://ftp.us.debian.org/debian/pool/main/x/x11proto-xinerama/x11proto-xinerama-dev_1.2.1-2_all.deb
 http://ftp.us.debian.org/debian/pool/main/z/zlib/zlib1g_1.2.8.dfsg-5_mipsel.deb
 http://ftp.us.debian.org/debian/pool/main/z/zlib/zlib1g-dev_1.2.8.dfsg-5_mipsel.deb
+http://security.debian.org/pool/updates/main/b/bluez/libbluetooth3_5.43-2+deb9u1_mipsel.deb
+http://security.debian.org/pool/updates/main/b/bluez/libbluetooth-dev_5.43-2+deb9u1_mipsel.deb
 http://security.debian.org/pool/updates/main/e/expat/libexpat1_2.2.0-2+deb9u1_mipsel.deb
 http://security.debian.org/pool/updates/main/e/expat/libexpat1-dev_2.2.0-2+deb9u1_mipsel.deb
+http://security.debian.org/pool/updates/main/g/gdk-pixbuf/libgdk-pixbuf2.0-0_2.36.5-2+deb9u1_mipsel.deb
+http://security.debian.org/pool/updates/main/g/gdk-pixbuf/libgdk-pixbuf2.0-dev_2.36.5-2+deb9u1_mipsel.deb
 http://security.debian.org/pool/updates/main/g/glibc/libc6_2.24-11+deb9u1_mipsel.deb
 http://security.debian.org/pool/updates/main/g/glibc/libc6-dev_2.24-11+deb9u1_mipsel.deb
 http://security.debian.org/pool/updates/main/g/gnutls28/libgnutls28-dev_3.5.8-5+deb9u1_mipsel.deb
@@ -209,6 +211,6 @@ http://security.debian.org/pool/updates/main/g/gnutls28/libgnutls30_3.5.8-5+deb9
 http://security.debian.org/pool/updates/main/g/gnutls28/libgnutls-dane0_3.5.8-5+deb9u1_mipsel.deb
 http://security.debian.org/pool/updates/main/g/gnutls28/libgnutls-openssl27_3.5.8-5+deb9u1_mipsel.deb
 http://security.debian.org/pool/updates/main/g/gnutls28/libgnutlsxx28_3.5.8-5+deb9u1_mipsel.deb
-http://security.debian.org/pool/updates/main/libg/libgcrypt20/libgcrypt20_1.7.6-2+deb9u1_mipsel.deb
-http://security.debian.org/pool/updates/main/libg/libgcrypt20/libgcrypt20-dev_1.7.6-2+deb9u1_mipsel.deb
-http://security.debian.org/pool/updates/main/l/linux/linux-libc-dev_4.9.30-2+deb9u2_mipsel.deb
+http://security.debian.org/pool/updates/main/libg/libgcrypt20/libgcrypt20_1.7.6-2+deb9u2_mipsel.deb
+http://security.debian.org/pool/updates/main/libg/libgcrypt20/libgcrypt20-dev_1.7.6-2+deb9u2_mipsel.deb
+http://security.debian.org/pool/updates/main/l/linux/linux-libc-dev_4.9.30-2+deb9u5_mipsel.deb
diff --git a/build/linux/sysroot_scripts/sysroot-creator-jessie.sh b/build/linux/sysroot_scripts/sysroot-creator-jessie.sh
index 8b668dc7141c..0a4cb2f16a0f 100755
--- a/build/linux/sysroot_scripts/sysroot-creator-jessie.sh
+++ b/build/linux/sysroot_scripts/sysroot-creator-jessie.sh
@@ -60,6 +60,8 @@ DEBIAN_PACKAGES="\
   libdbus-1-3
   libdbus-1-dev
   libdbus-glib-1-2
+  libdconf-dev
+  libdconf1
   libdrm2
   libdrm-dev
   libdrm-nouveau2
diff --git a/build/linux/sysroot_scripts/sysroot-creator-stretch.sh b/build/linux/sysroot_scripts/sysroot-creator-stretch.sh
index 5bf572bf3408..81c314787af9 100755
--- a/build/linux/sysroot_scripts/sysroot-creator-stretch.sh
+++ b/build/linux/sysroot_scripts/sysroot-creator-stretch.sh
@@ -63,6 +63,8 @@ DEBIAN_PACKAGES="\
   libdbus-1-3
   libdbus-1-dev
   libdbus-glib-1-2
+  libdconf-dev
+  libdconf1
   libdrm2
   libdrm-dev
   libdrm-amdgpu1
diff --git a/build/linux/sysroot_scripts/sysroots.json b/build/linux/sysroot_scripts/sysroots.json
index c11bd458a1cf..14c6716cf06d 100644
--- a/build/linux/sysroot_scripts/sysroots.json
+++ b/build/linux/sysroot_scripts/sysroots.json
@@ -1,62 +1,68 @@
 {
     "jessie_amd64": {
-        "Revision": "e6ac45fd042859d2240e5d432c521df01ad5b7bf",
-        "Sha1Sum": "249cfcb2fa8967d18d4a52d6a688a62f09a05985",
+        "Revision": "823bd3badb80d935f8029a06218d88dcf380ca82",
+        "Sha1Sum": "cdf4b63d578b78491a3dd527730b4abb6ad3c0ef",
         "SysrootDir": "debian_jessie_amd64-sysroot",
         "Tarball": "debian_jessie_amd64_sysroot.tar.xz"
     },
     "jessie_arm": {
-        "Revision": "e6ac45fd042859d2240e5d432c521df01ad5b7bf",
-        "Sha1Sum": "be4fd14f0390590f7e9202953ddbd6e6a56038c1",
+        "Revision": "823bd3badb80d935f8029a06218d88dcf380ca82",
+        "Sha1Sum": "c5eeee2f84f07c76f4d46849de561b7ec3ca7638",
         "SysrootDir": "debian_jessie_arm-sysroot",
         "Tarball": "debian_jessie_arm_sysroot.tar.xz"
     },
     "jessie_arm64": {
-        "Revision": "e6ac45fd042859d2240e5d432c521df01ad5b7bf",
-        "Sha1Sum": "977a723487aee40507f04d13f68e6d109d67a92c",
+        "Revision": "823bd3badb80d935f8029a06218d88dcf380ca82",
+        "Sha1Sum": "ed372720bf897cd5565e552c6a7f0d510b572c44",
         "SysrootDir": "debian_jessie_arm64-sysroot",
         "Tarball": "debian_jessie_arm64_sysroot.tar.xz"
     },
     "jessie_i386": {
-        "Revision": "e6ac45fd042859d2240e5d432c521df01ad5b7bf",
-        "Sha1Sum": "4f418577fc17023d78872f5e5ebe66b1c7009f69",
+        "Revision": "823bd3badb80d935f8029a06218d88dcf380ca82",
+        "Sha1Sum": "3c9ce5efc973b5342b04e9c9c97551ce8ce2c76b",
         "SysrootDir": "debian_jessie_i386-sysroot",
         "Tarball": "debian_jessie_i386_sysroot.tar.xz"
     },
     "jessie_mips": {
-        "Revision": "e6ac45fd042859d2240e5d432c521df01ad5b7bf",
-        "Sha1Sum": "67da4cf69e87719d3ca8427771c927b6af95ffba",
+        "Revision": "823bd3badb80d935f8029a06218d88dcf380ca82",
+        "Sha1Sum": "567a036527d98388271c097f225212080cf814a2",
         "SysrootDir": "debian_jessie_mips-sysroot",
         "Tarball": "debian_jessie_mips_sysroot.tar.xz"
     },
     "stretch_amd64": {
-        "Revision": "7d1a17d2a858e32ce6b8a237d603019a2e3ac0b0",
-        "Sha1Sum": "4fd5bacc951f648462702952ce14063b306ce6ba",
+        "Revision": "823bd3badb80d935f8029a06218d88dcf380ca82",
+        "Sha1Sum": "83e93938092c750565435663049272a50b657c3e",
         "SysrootDir": "debian_stretch_amd64-sysroot",
         "Tarball": "debian_stretch_amd64_sysroot.tar.xz"
     },
     "stretch_arm": {
-        "Revision": "7d1a17d2a858e32ce6b8a237d603019a2e3ac0b0",
-        "Sha1Sum": "11e30b748de602ece73687ec0bf972e662a31f3e",
+        "Revision": "823bd3badb80d935f8029a06218d88dcf380ca82",
+        "Sha1Sum": "4f61e21b003d94d6004c07390e663d3f8e638fc5",
         "SysrootDir": "debian_stretch_arm-sysroot",
         "Tarball": "debian_stretch_arm_sysroot.tar.xz"
     },
     "stretch_arm64": {
-        "Revision": "7d1a17d2a858e32ce6b8a237d603019a2e3ac0b0",
-        "Sha1Sum": "f863c58810c8b4e9b556ffc7fc3980f9e3ba6fc2",
+        "Revision": "823bd3badb80d935f8029a06218d88dcf380ca82",
+        "Sha1Sum": "f812db22eba59fe4890b06fca3ddcc109f616de0",
         "SysrootDir": "debian_stretch_arm64-sysroot",
         "Tarball": "debian_stretch_arm64_sysroot.tar.xz"
     },
     "stretch_i386": {
-        "Revision": "7d1a17d2a858e32ce6b8a237d603019a2e3ac0b0",
-        "Sha1Sum": "32362c929300daad99c071ed80b697a618e76c79",
+        "Revision": "823bd3badb80d935f8029a06218d88dcf380ca82",
+        "Sha1Sum": "f45dd3370b5d40e3e8489766f5ec6c015647c1d6",
         "SysrootDir": "debian_stretch_i386-sysroot",
         "Tarball": "debian_stretch_i386_sysroot.tar.xz"
     },
     "stretch_mips": {
-        "Revision": "7d1a17d2a858e32ce6b8a237d603019a2e3ac0b0",
-        "Sha1Sum": "182f5c1d3347d54894ef31e273e2859d4ddc212d",
+        "Revision": "823bd3badb80d935f8029a06218d88dcf380ca82",
+        "Sha1Sum": "2ea51db835e777858befcbb6e7cfb42f3f655f9f",
         "SysrootDir": "debian_stretch_mips-sysroot",
         "Tarball": "debian_stretch_mips_sysroot.tar.xz"
+    },
+    "stretch_mips64el": {
+        "Revision": "823bd3badb80d935f8029a06218d88dcf380ca82",
+        "Sha1Sum": "1d35dfdc1573a7a2a6237160a8dd122dc7a5fff8",
+        "SysrootDir": "debian_stretch_mips64el-sysroot",
+        "Tarball": "debian_stretch_mips64el_sysroot.tar.xz"
     }
 }
