b4ac34edef3a7552cb38c938f00eb45885047044
diff --git a/build/linux/sysroot_scripts/install-sysroot.py b/build/linux/sysroot_scripts/install-sysroot.py
index 181ac9368fcd..de4eead5efe1 100755
--- a/build/linux/sysroot_scripts/install-sysroot.py
+++ b/build/linux/sysroot_scripts/install-sysroot.py
@@ -169,7 +169,8 @@ def InstallDefaultSysrootForArch(target_arch):
 
 
 def InstallSysroot(target_platform, target_arch):
-  # The sysroot directory should match the one specified in build/common.gypi.
+  # The sysroot directory should match the one specified in
+  # build/config/sysroot.gni.
   # TODO(thestig) Consider putting this elsewhere to avoid having to recreate
   # it on every build.
   linux_dir = os.path.dirname(SCRIPT_DIR)
@@ -191,8 +192,6 @@ def InstallSysroot(target_platform, target_arch):
   if os.path.exists(stamp):
     with open(stamp) as s:
       if s.read() == url:
-        print '%s %s sysroot image already up to date: %s' % \
-            (target_platform, target_arch, sysroot)
         return
 
   print 'Installing Debian %s %s root image: %s' % \
