diff --git a/base/process/launch.cc b/base/process/launch.cc
index f09317d..1785aea 100644
--- a/base/process/launch.cc
+++ b/base/process/launch.cc
@@ -27,7 +27,7 @@ LaunchOptions::LaunchOptions()
       new_process_group(false)
 #if defined(OS_LINUX)
       , clone_flags(0)
-      , allow_new_privs(false)
+      , allow_new_privs(true)
       , kill_on_parent_death(false)
 #endif  // OS_LINUX
 #if defined(OS_POSIX)
