diff --git a/third_party/WebKit/Source/bindings/core/v8/V8RecursionScope.h b/third_party/WebKit/Source/bindings/core/v8/V8RecursionScope.h
index dd9a51f..22299c3 100644
--- a/third_party/WebKit/Source/bindings/core/v8/V8RecursionScope.h
+++ b/third_party/WebKit/Source/bindings/core/v8/V8RecursionScope.h
@@ -56,7 +56,6 @@ namespace blink {
 //
 // http://www.whatwg.org/specs/web-apps/current-work/#perform-a-microtask-checkpoint
 class CORE_EXPORT V8RecursionScope {
-    STACK_ALLOCATED();
 public:
     explicit V8RecursionScope(v8::Isolate* isolate)
         : m_isolate(isolate)
diff --git a/third_party/WebKit/Source/web/WebScopedMicrotaskSuppression.cpp b/third_party/WebKit/Source/web/WebScopedMicrotaskSuppression.cpp
index a719a9c..752fd82 100644
--- a/third_party/WebKit/Source/web/WebScopedMicrotaskSuppression.cpp
+++ b/third_party/WebKit/Source/web/WebScopedMicrotaskSuppression.cpp
@@ -36,6 +36,27 @@
 
 namespace blink {
 
+class WebScopedRunV8Script::Impl {
+public:
+    Impl(v8::Isolate* isolate)
+        : m_scope(isolate)
+    {
+    }
+
+private:
+    V8RecursionScope m_scope;
+};
+
+void WebScopedRunV8Script::initialize()
+{
+    m_impl.reset(new Impl(v8::Isolate::GetCurrent()));
+}
+
+void WebScopedRunV8Script::reset()
+{
+    m_impl.reset(0);
+}
+
 #if ENABLE(ASSERT)
 class WebScopedMicrotaskSuppression::Impl : public V8RecursionScope::MicrotaskSuppression {
 public:
diff --git a/third_party/WebKit/public/web/WebScopedMicrotaskSuppression.h b/third_party/WebKit/public/web/WebScopedMicrotaskSuppression.h
index dd96fc6..2644834 100644
--- a/third_party/WebKit/public/web/WebScopedMicrotaskSuppression.h
+++ b/third_party/WebKit/public/web/WebScopedMicrotaskSuppression.h
@@ -71,6 +71,23 @@ private:
     WebPrivateOwnPtr<Impl> m_impl;
 };
 
+// This class wraps V8RecursionScope, any direct call into V8 should be guarded
+// inside an instance of this class, otherwise microtask checkpoint will be
+// executed too early.
+// See http://git.io/vOFN5.
+class WebScopedRunV8Script {
+public:
+    explicit WebScopedRunV8Script() { initialize(); }
+    ~WebScopedRunV8Script() { reset(); }
+
+private:
+    BLINK_EXPORT void initialize();
+    BLINK_EXPORT void reset();
+
+    class Impl;
+    WebPrivateOwnPtr<Impl> m_impl;
+};
+
 } // namespace blink
 
 #endif
