diff --git a/third_party/WebKit/Source/core/editing/EditingBehavior.h b/third_party/WebKit/Source/core/editing/EditingBehavior.h
index 3f098ef..01f1ad4 100644
--- a/third_party/WebKit/Source/core/editing/EditingBehavior.h
+++ b/third_party/WebKit/Source/core/editing/EditingBehavior.h
@@ -85,7 +85,7 @@ class CORE_EXPORT EditingBehavior {
   // On Mac and Windows, pressing backspace (when it isn't handled otherwise)
   // should navigate back.
   bool shouldNavigateBackOnBackspace() const {
-    return m_type != EditingUnixBehavior && m_type != EditingAndroidBehavior;
+    return false;
   }
 
   // On Mac, selecting backwards by word/line from the middle of a word/line,
