diff --git a/third_party/WebKit/Source/core/frame/LocalFrame.cpp b/third_party/WebKit/Source/core/frame/LocalFrame.cpp
index 97a5393d40b4..04f88dd2ea15 100644
--- a/third_party/WebKit/Source/core/frame/LocalFrame.cpp
+++ b/third_party/WebKit/Source/core/frame/LocalFrame.cpp
@@ -435,10 +435,6 @@ void LocalFrame::Detach(FrameDetachType type) {
   loader_.StopAllLoaders();
   loader_.Detach();
   GetDocument()->Shutdown();
-  // This is the earliest that scripting can be disabled:
-  // - FrameLoader::detach() can fire XHR abort events
-  // - Document::shutdown()'s deferred widget updates can run script.
-  ScriptForbiddenScope forbid_script;
   if (!Client())
     return;
 
@@ -446,6 +442,10 @@ void LocalFrame::Detach(FrameDetachType type) {
   // Notify ScriptController that the frame is closing, since its cleanup ends
   // up calling back to LocalFrameClient via WindowProxy.
   GetScriptController().ClearForClose();
+  // This is the earliest that scripting can be disabled:
+  // - FrameLoader::detach() can fire XHR abort events
+  // - Document::shutdown()'s deferred widget updates can run script.
+  ScriptForbiddenScope forbid_script;
   SetView(nullptr);
 
   page_->GetEventHandlerRegistry().DidRemoveAllEventHandlers(*DomWindow());
