diff --git a/third_party/boringssl/boringssl.gyp b/third_party/boringssl/boringssl.gyp
index d0b716a..cfe0bb6 100644
--- a/third_party/boringssl/boringssl.gyp
+++ b/third_party/boringssl/boringssl.gyp
@@ -3,10 +3,13 @@
 # found in the LICENSE file.
 
 {
+  'variables': {
+    'boringssl_component%': 'shared_library',
+  },
   'targets': [
     {
       'target_name': 'boringssl',
-      'type': '<(component)',
+      'type': '<(boringssl_component)',
       'includes': [
         'boringssl.gypi',
       ],
@@ -21,7 +24,7 @@
       # https://crbug.com/429039
       'msvs_disabled_warnings': [ 4267, ],
       'conditions': [
-        ['component == "shared_library"', {
+        ['boringssl_component == "shared_library"', {
           'defines': [
             'BORINGSSL_SHARED_LIBRARY',
           ],
@@ -93,7 +96,7 @@
           'src/include',
         ],
         'conditions': [
-          ['component == "shared_library"', {
+          ['boringssl_component == "shared_library"', {
             'defines': [
               'BORINGSSL_SHARED_LIBRARY',
             ],
