diff --git a/build/config/BUILDCONFIG.gn b/build/config/BUILDCONFIG.gn
index f354f17..1c86ff0 100644
--- a/build/config/BUILDCONFIG.gn
+++ b/build/config/BUILDCONFIG.gn
@@ -123,6 +123,8 @@ if (current_os == "") {
 #   even if the value is overridden, which is wasteful. See first bullet.
 
 declare_args() {
+  is_electron_build = false
+
   # Set to enable the official build level of optimization. This has nothing
   # to do with branding, but enables an additional level of optimization above
   # release (!is_debug). This might be better expressed as a tri-state
@@ -480,6 +482,7 @@ default_compiler_configs = [
   "//build/config/compiler:no_rtti",
   "//build/config/compiler:runtime_library",
   "//build/config/sanitizers:default_sanitizer_flags",
+  "//chromiumcontent:mas_build",
 ]
 if (is_win) {
   default_compiler_configs += [
diff --git a/third_party/WebKit/Source/platform/BUILD.gn b/third_party/WebKit/Source/platform/BUILD.gn
index 453c14f..55dd9f3 100644
--- a/third_party/WebKit/Source/platform/BUILD.gn
+++ b/third_party/WebKit/Source/platform/BUILD.gn
@@ -171,6 +171,7 @@ action("character_data") {
 }
 
 executable("character_data_generator") {
+  configs += [ "//chromiumcontent:build_time_executable" ]
   sources = [
     "text/CharacterPropertyDataGenerator.cpp",
     "text/CharacterPropertyDataGenerator.h",
diff --git a/third_party/widevine/cdm/BUILD.gn b/third_party/widevine/cdm/BUILD.gn
index 8b40b1a..1028293 100644
--- a/third_party/widevine/cdm/BUILD.gn
+++ b/third_party/widevine/cdm/BUILD.gn
@@ -105,7 +105,7 @@ if (widevine_cdm_binary_files != []) {
       "//build/config/sanitizers:deps",
     ]
 
-    if (is_mac) {
+    if (is_mac && !is_electron_build) {
       ldflags = [
         # Not to strip important symbols by -Wl,-dead_strip.
         "-Wl,-exported_symbol,_PPP_GetInterface",
