diff --git a/base/allocator/BUILD.gn b/base/allocator/BUILD.gn
index 8cdb06161f5c..6f938d6be67b 100644
--- a/base/allocator/BUILD.gn
+++ b/base/allocator/BUILD.gn
@@ -10,12 +10,13 @@ declare_args() {
   # Provide a way to force disable debugallocation in Debug builds,
   # e.g. for profiling (it's more rare to profile Debug builds,
   # but people sometimes need to do that).
-  enable_debugallocation = is_debug
+  enable_debugallocation = false
 }
 
 # The Windows-only allocator shim is only enabled for Release static builds, and
 # is mutually exclusive with the generalized shim.
 win_use_allocator_shim = is_win && !is_component_build && !is_debug &&
+                         !is_electron_build &&
                          !use_experimental_allocator_shim && !is_asan
 
 # This "allocator" meta-target will forward to the default allocator according
diff --git a/build/config/BUILDCONFIG.gn b/build/config/BUILDCONFIG.gn
index d055e70..c76ca27 100644
--- a/build/config/BUILDCONFIG.gn
+++ b/build/config/BUILDCONFIG.gn
@@ -123,6 +123,8 @@ if (current_os == "") {
 #   even if the value is overridden, which is wasteful. See first bullet.
 
 declare_args() {
+  is_electron_build = false
+
   # Set to enable the official build level of optimization. This has nothing
   # to do with branding, but enables an additional level of optimization above
   # release (!is_debug). This might be better expressed as a tri-state
@@ -483,6 +485,7 @@ default_compiler_configs = [
   "//build/config/compiler:no_rtti",
   "//build/config/compiler:runtime_library",
   "//build/config/sanitizers:default_sanitizer_flags",
+  "//chromiumcontent:mas_build",
 ]
 if (is_win) {
   default_compiler_configs += [
diff --git a/build/config/win/BUILD.gn b/build/config/win/BUILD.gn
index 5e1f7fcc4d83..15723e6b7184 100644
--- a/build/config/win/BUILD.gn
+++ b/build/config/win/BUILD.gn
@@ -276,7 +276,7 @@ config("common_linker_setup") {
 # See https://msdn.microsoft.com/en-us/library/2kzt1wy3.aspx for a reference of
 # what each value does.
 config("default_crt") {
-  if (is_component_build) {
+  if (is_component_build || is_electron_build) {
     # Component mode: dynamic CRT. Since the library is shared, it requires
     # exceptions or will give errors about things not matching, so keep
     # exceptions on.
diff --git a/third_party/WebKit/Source/platform/BUILD.gn b/third_party/WebKit/Source/platform/BUILD.gn
index 15bf689..9cdc19d 100644
--- a/third_party/WebKit/Source/platform/BUILD.gn
+++ b/third_party/WebKit/Source/platform/BUILD.gn
@@ -160,6 +160,7 @@ action("character_data") {
 }
 
 executable("character_data_generator") {
+  configs += [ "//chromiumcontent:build_time_executable" ]
   sources = [
     "text/CharacterPropertyDataGenerator.cpp",
     "text/CharacterPropertyDataGenerator.h",
diff --git a/third_party/widevine/cdm/BUILD.gn b/third_party/widevine/cdm/BUILD.gn
index dda9799..2bf3f57 100644
--- a/third_party/widevine/cdm/BUILD.gn
+++ b/third_party/widevine/cdm/BUILD.gn
@@ -153,7 +153,7 @@ if (widevine_cdm_binary_files != []) {
       "//build/config/sanitizers:deps",
     ]
 
-    if (is_mac) {
+    if (is_mac && !is_electron_build) {
       ldflags = [
         # Not to strip important symbols by -Wl,-dead_strip.
         "-Wl,-exported_symbol,_PPP_GetInterface",
