Index: webkit/plugins/ppapi/content_decryptor_delegate.cc
===================================================================
--- webkit/plugins/ppapi/content_decryptor_delegate.cc	(revision 194718)
+++ webkit/plugins/ppapi/content_decryptor_delegate.cc	(working copy)
@@ -285,6 +285,9 @@
       weak_this_(weak_ptr_factory_.GetWeakPtr()) {
 }
 
+ContentDecryptorDelegate::~ContentDecryptorDelegate() {
+}
+
 void ContentDecryptorDelegate::Initialize(const std::string& key_system) {
   // TODO(ddorwin): Add an Initialize method to PPP_ContentDecryptor_Private.
   DCHECK(!key_system.empty());
   key_system_ = key_system;
Index: webkit/plugins/ppapi/content_decryptor_delegate.h
===================================================================
--- webkit/plugins/ppapi/content_decryptor_delegate.h	(revision 194718)
+++ webkit/plugins/ppapi/content_decryptor_delegate.h	(working copy)
@@ -38,6 +38,7 @@
   ContentDecryptorDelegate(
       PP_Instance pp_instance,
       const PPP_ContentDecryptor_Private* plugin_decryption_interface);
+  ~ContentDecryptorDelegate();
 
   void Initialize(const std::string& key_system);

