Index: content/content_shell.gypi
===================================================================
--- content/content_shell.gypi	(revision 274914)
+++ content/content_shell.gypi	(working copy)
@@ -492,11 +492,7 @@
             ],
             'conditions': [
               ['OS!="android"', {
-                'variables': {
-                  'pak_inputs': [
-                    '<(SHARED_INTERMEDIATE_DIR)/webkit/devtools_resources.pak',
-                  ],
-                },
+                'pak_inputs': ['<(SHARED_INTERMEDIATE_DIR)/webkit/devtools_resources.pak',],
                 'pak_output': '<(PRODUCT_DIR)/content_shell.pak',
               }, {
                 'pak_output': '<(PRODUCT_DIR)/content_shell/assets/content_shell.pak',
