diff --git a/content/BUILD.gn b/content/BUILD.gn
index d332c24..a12bc2c 100644
--- a/content/BUILD.gn
+++ b/content/BUILD.gn
@@ -139,7 +139,7 @@ source_set("export") {
 # In the GYP build, this file is listed in several targets. In GN just have
 # those targets depend on this one. This can be depended on for any
 # platform for simplicity, and is a no-op on non-Windows.
-source_set("sandbox_helper_win") {
+static_library("sandbox_helper_win") {
   if (is_win) {
     sources = [
       "app/sandbox_helper_win.cc",
