Index: webkit/dom_storage/dom_storage_host.h
===================================================================
--- webkit/dom_storage/dom_storage_host.h	(revision 194718)
+++ webkit/dom_storage/dom_storage_host.h	(working copy)
@@ -64,6 +64,8 @@
 
   scoped_refptr<DomStorageContext> context_;
   AreaMap connections_;
+
+  DISALLOW_COPY_AND_ASSIGN(DomStorageHost);
 };
 
 }  // namespace dom_storage
