Index: ui/events/win/events_win.cc
===================================================================
--- ui/events/win/events_win.cc	(revision 274914)
+++ ui/events/win/events_win.cc	(working copy)
@@ -236,7 +236,7 @@
     native_point.y = GET_Y_LPARAM(native_event.lParam);
   }
   ScreenToClient(native_event.hwnd, &native_point);
-  return gfx::win::ScreenToDIPPoint(gfx::Point(native_point));
+  return gfx::Point(native_point);
 }
 
 gfx::Point EventSystemLocationFromNative(
