diff --git a/pdf/pdf.cc b/pdf/pdf.cc
index 1f580e4..939b54d 100644
--- a/pdf/pdf.cc
+++ b/pdf/pdf.cc
@@ -100,7 +100,12 @@ const void* PPP_GetInterface(const char* interface_name) {
   return pp::Module::Get()->GetPluginInterface(interface_name);
 }
 
+}  // namespace chrome_pdf
+
+extern "C" {
+
 #if defined(OS_WIN)
+PDF_EXPORT
 bool RenderPDFPageToDC(const void* pdf_buffer,
                                  int buffer_size,
                                  int page_number,
@@ -137,6 +142,7 @@ bool RenderPDFPageToDC(const void* pdf_buffer,
 
 #endif  // OS_WIN
 
+PDF_EXPORT
 bool GetPDFDocInfo(const void* pdf_buffer,
                    int buffer_size, int* page_count,
                    double* max_page_width) {
diff --git a/pdf/pdf.gyp b/pdf/pdf.gyp
index bb6a36f..508ac16 100644
--- a/pdf/pdf.gyp
+++ b/pdf/pdf.gyp
@@ -6,7 +6,7 @@
   'targets': [
     {
       'target_name': 'pdf',
-      'type': 'static_library',
+      'type': 'loadable_module',
       'dependencies': [
         '../base/base.gyp:base',
         '../components/components.gyp:ui_zoom',
@@ -15,6 +15,7 @@
         '../net/net.gyp:net',
         '../ppapi/ppapi.gyp:ppapi_cpp_objects',
         '../ppapi/ppapi.gyp:ppapi_internal_module',
+        '../printing/printing.gyp:printing',
         '../third_party/pdfium/pdfium.gyp:pdfium',
       ],
       'ldflags': [ '-L<(PRODUCT_DIR)',],
diff --git a/pdf/pdf.h b/pdf/pdf.h
index 37e72e5..57d673c 100644
--- a/pdf/pdf.h
+++ b/pdf/pdf.h
@@ -8,6 +8,12 @@
 #include "ppapi/c/ppb.h"
 #include "ppapi/cpp/module.h"
 
+#if defined(WIN32)
+#define PDF_EXPORT __declspec(dllexport)
+#else
+#define PDF_EXPORT
+#endif // defined(WIN32)
+
 namespace chrome_pdf {
 
 class PDFModule : public pp::Module {
@@ -25,6 +31,7 @@ int PPP_InitializeModule(PP_Module module_id,
 void PPP_ShutdownModule();
 const void* PPP_GetInterface(const char* interface_name);
 
+#if 0
 #if defined(OS_WIN)
 // |pdf_buffer| is the buffer that contains the entire PDF document to be
 //     rendered.
@@ -107,6 +114,7 @@ bool RenderPDFPageToBitmap(const void* pdf_buffer,
                            int bitmap_height,
                            int dpi,
                            bool autorotate);
+#endif
 
 }  // namespace chrome_pdf
 
