diff --git a/tools/gn/target_generator.cc b/tools/gn/target_generator.cc
index 7fe64fd..dd4bfa1 100644
--- a/tools/gn/target_generator.cc
+++ b/tools/gn/target_generator.cc
@@ -128,7 +128,10 @@ void TargetGenerator::GenerateTarget(Scope* scope,
     generator.Run();
   } else if (output_type == functions::kSourceSet) {
     BinaryTargetGenerator generator(target.get(), scope, function_call,
-                                    Target::SOURCE_SET, err);
+                                    target->sources().empty() ?
+                                    Target::SOURCE_SET :
+                                    Target::STATIC_LIBRARY,
+                                    err);
     generator.Run();
   } else if (output_type == functions::kStaticLibrary) {
     BinaryTargetGenerator generator(target.get(), scope, function_call,
