Index: components/webdata.gypi
===================================================================
--- components/webdata.gypi	(revision 226629)
+++ components/webdata.gypi	(working copy)
@@ -25,6 +25,9 @@
         'webdata/encryptor/ie7_password.h',
       ],
       'conditions': [
+        ['OS == "linux"', {
+          'standalone_static_library': 1,
+        }],
         ['OS!="win"', {
           'sources!': [
             'webdata/encryptor/ie7_password.cc'
Index: testing/gtest.gyp
===================================================================
--- testing/gtest.gyp	(revision 226629)
+++ testing/gtest.gyp	(working copy)
@@ -51,6 +51,9 @@
         'gtest_prod',
       ],
       'conditions': [
+        ['OS == "linux"', {
+          'standalone_static_library': 1,
+        }],
         ['OS == "mac" or OS == "ios"', {
           'sources': [
             'gtest_mac.h',
