diff --git a/chrome/chrome_resources.gyp b/chrome/chrome_resources.gyp
index 6edb819..6f9ff74 100644
--- a/chrome/chrome_resources.gyp
+++ b/chrome/chrome_resources.gyp
@@ -656,16 +656,16 @@
       # GN version: //chrome/test/data/resources:webui_test_resources
       'target_name': 'webui_test_resources',
       'type': 'none',
-      'actions': [
-        {
-          'action_name': 'generate_webui_test_resources',
-          'variables': {
-            'grit_grd_file': 'test/data/webui_test_resources.grd',
-          },
-          'includes': [ '../build/grit_action.gypi' ],
-        },
-      ],
-      'includes': [ '../build/grit_target.gypi' ],
+      # 'actions': [
+      #   {
+      #     'action_name': 'generate_webui_test_resources',
+      #     'variables': {
+      #       'grit_grd_file': 'test/data/webui_test_resources.grd',
+      #     },
+      #     'includes': [ '../build/grit_action.gypi' ],
+      #   },
+      # ],
+      # 'includes': [ '../build/grit_target.gypi' ],
     },
     {
       # GN version: //chrome:browser_tests_pak
