diff --git a/base/mac/sdk_forward_declarations.h b/base/mac/sdk_forward_declarations.h
index 363ada1..8cc3708 100644
--- a/base/mac/sdk_forward_declarations.h
+++ b/base/mac/sdk_forward_declarations.h
@@ -545,8 +545,7 @@ BASE_EXPORT extern "C" void NSAccessibilityPostNotificationWithUserInfo(
 // These declarations and definitions will not be necessary once Chrome no
 // longer runs on OSX 10.6.
 // ----------------------------------------------------------------------------
-#if defined(MAC_OS_X_VERSION_10_9) && \
-    MAC_OS_X_VERSION_MIN_REQUIRED <= MAC_OS_X_VERSION_10_6
+#if defined(MAC_OS_X_VERSION_10_9)
 typedef enum {
   kCWSecurityModeOpen = 0,
   kCWSecurityModeWEP,
