diff --git a/content/browser/renderer_host/render_message_filter.cc b/content/browser/renderer_host/render_message_filter.cc
index 29ab868..fe60d60 100644
--- a/content/browser/renderer_host/render_message_filter.cc
+++ b/content/browser/renderer_host/render_message_filter.cc
@@ -254,6 +254,7 @@ void RenderMessageFilter::OnCreateWindow(
           params.opener_top_level_frame_url,
           params.opener_security_origin,
           params.window_container_type,
+          params.frame_name,
           params.target_url,
           params.referrer,
           params.disposition,
diff --git a/content/public/browser/content_browser_client.cc b/content/public/browser/content_browser_client.cc
index 8ec8c8c..94d5979 100644
--- a/content/public/browser/content_browser_client.cc
+++ b/content/public/browser/content_browser_client.cc
@@ -294,6 +294,7 @@ bool ContentBrowserClient::CanCreateWindow(
     const GURL& opener_top_level_frame_url,
     const GURL& source_origin,
     WindowContainerType container_type,
+    const std::string& frame_name,
     const GURL& target_url,
     const Referrer& referrer,
     WindowOpenDisposition disposition,
diff --git a/content/public/browser/content_browser_client.h b/content/public/browser/content_browser_client.h
index dd3f0e6..623906a 100644
--- a/content/public/browser/content_browser_client.h
+++ b/content/public/browser/content_browser_client.h
@@ -532,6 +532,7 @@ class CONTENT_EXPORT ContentBrowserClient {
                                const GURL& opener_top_level_frame_url,
                                const GURL& source_origin,
                                WindowContainerType container_type,
+                               const std::string& frame_name,
                                const GURL& target_url,
                                const Referrer& referrer,
                                WindowOpenDisposition disposition,
