diff --git a/content/browser/renderer_host/render_widget_host_view_mac.mm b/content/browser/renderer_host/render_widget_host_view_mac.mm
index 8143dfa..a033f08 100644
--- a/content/browser/renderer_host/render_widget_host_view_mac.mm
+++ b/content/browser/renderer_host/render_widget_host_view_mac.mm
@@ -2515,6 +2515,18 @@ void RenderWidgetHostViewMac::OnDisplayMetricsChanged(
 }
 
 - (BOOL)acceptsFirstMouse:(NSEvent*)theEvent {
+  const SEL selector = @selector(acceptsFirstMouse:);
+  if ([self.window.delegate respondsToSelector:selector]) {
+    NSInvocation* invocation = [NSInvocation
+        invocationWithMethodSignature:[[self.window.delegate class]
+            instanceMethodSignatureForSelector:selector]];
+    [invocation setTarget:self.window.delegate];
+    [invocation setSelector:selector];
+    [invocation invoke];
+    BOOL accept;
+    [invocation getReturnValue:&accept];
+    return accept;
+  }
   return [self acceptsMouseEventsWhenInactive];
 }
 
@@ -2652,6 +2664,7 @@ void RenderWidgetHostViewMac::OnDisplayMetricsChanged(
   if (EventIsReservedBySystem(theEvent))
     return NO;
 
+#if 0
   // If we return |NO| from this function, cocoa will send the key event to
   // the menu and only if the menu does not process the event to |keyDown:|. We
   // want to send the event to a renderer _before_ sending it to the menu, so
@@ -2665,6 +2678,7 @@ void RenderWidgetHostViewMac::OnDisplayMetricsChanged(
     DCHECK(![[NSApp mainMenu] performKeyEquivalent:theEvent]);
     return NO;
   }
+#endif
 
   // Command key combinations are sent via performKeyEquivalent rather than
   // keyDown:. We just forward this on and if WebCore doesn't want to handle
