diff --git a/content/public/browser/resource_request_details.cc b/content/public/browser/resource_request_details.cc
index ad5f47e..4e87c72 100644
--- a/content/public/browser/resource_request_details.cc
+++ b/content/public/browser/resource_request_details.cc
@@ -6,6 +6,7 @@
 
 #include "content/public/browser/resource_request_info.h"
 #include "net/http/http_response_headers.h"
+#include "net/http/http_util.h"
 #include "net/url_request/url_request.h"
 
 namespace content {
@@ -29,6 +30,11 @@ ResourceRequestDetails::ResourceRequestDetails(const net::URLRequest* request,
       request->response_info().headers.get() ?
           request->response_info().headers.get()->response_code() : -1;
   origin_child_id = info->GetChildID();
+  if (request->response_info().headers.get()) {
+    std::string raw_headers = request->response_info().headers->raw_headers();
+    headers = new net::HttpResponseHeaders(net::HttpUtil::AssembleRawHeaders(
+        raw_headers.c_str(), raw_headers.size()));
+  }
 }
 
 ResourceRequestDetails::~ResourceRequestDetails() {}
diff --git a/content/public/browser/resource_request_details.h b/content/public/browser/resource_request_details.h
index f776f4d..227439e 100644
--- a/content/public/browser/resource_request_details.h
+++ b/content/public/browser/resource_request_details.h
@@ -7,9 +7,11 @@
 
 #include <string>
 
+#include "base/memory/ref_counted.h"
 #include "content/public/common/resource_type.h"
 #include "net/base/host_port_pair.h"
 #include "net/cert/cert_status_flags.h"
+#include "net/http/http_response_headers.h"
 #include "net/url_request/url_request_status.h"
 #include "url/gurl.h"
 
@@ -43,6 +45,7 @@ struct ResourceRequestDetails {
   // HTTP response code. See HttpResponseHeaders::response_code().
   // -1 if there are no response headers yet.
   int http_response_code;
+  scoped_refptr<net::HttpResponseHeaders> headers;
 };
 
 // Details about a redirection of a resource request.
