Index: cc/resources/resource_update.h
===================================================================
--- cc/resources/resource_update.h	(revision 202921)
+++ cc/resources/resource_update.h	(working copy)
@@ -7,12 +7,12 @@
 
 #include "base/memory/ref_counted.h"
 #include "cc/base/cc_export.h"
+#include "third_party/skia/include/core/SkCanvas.h"
 #include "skia/ext/refptr.h"
 #include "ui/gfx/rect.h"
 #include "ui/gfx/vector2d.h"
 
 class SkBitmap;
-class SkCanvas;
 
 namespace cc {
 
