Index: sandbox/win/sandbox_win.gypi
===================================================================
--- sandbox/win/sandbox_win.gypi	(revision 194718)
+++ sandbox/win/sandbox_win.gypi	(working copy)
@@ -204,6 +204,21 @@
       ],
     },
     {
+      'target_name': 'sandbox_static',
+      'type': 'static_library',
+      'variables': {
+        'sandbox_windows_target': 1,
+      },
+      'dependencies': [
+        '../testing/gtest.gyp:gtest',
+        '../base/base.gyp:base',
+        '../base/base.gyp:base_static',
+      ],
+      'include_dirs': [
+        '../..',
+      ],
+    },
+    {
       'target_name': 'sbox_integration_tests',
       'type': 'executable',
       'dependencies': [
