diff --git a/sandbox/win/sandbox_win.gypi b/sandbox/win/sandbox_win.gypi
index 7d9cf94..ddd40c3 100644
--- a/sandbox/win/sandbox_win.gypi
+++ b/sandbox/win/sandbox_win.gypi
@@ -210,6 +210,21 @@
       ],
     },
     {
+      'target_name': 'sandbox_static',
+      'type': 'static_library',
+      'variables': {
+        'sandbox_windows_target': 1,
+      },
+      'dependencies': [
+        '../testing/gtest.gyp:gtest',
+        '../base/base.gyp:base',
+        '../base/base.gyp:base_static',
+      ],
+      'include_dirs': [
+        '../..',
+      ],
+    },
+    {
       'target_name': 'sbox_integration_tests',
       'type': 'executable',
       'dependencies': [
