diff --git a/sandbox/linux/services/credentials.cc b/sandbox/linux/services/credentials.cc
index 0e569fa..cada9e58 100644
--- a/sandbox/linux/services/credentials.cc
+++ b/sandbox/linux/services/credentials.cc
@@ -327,8 +327,10 @@ pid_t Credentials::ForkAndDropCapabilitiesInChild() {
     return pid;
   }
 
+#if 0
   // Since we just forked, we are single threaded.
   PCHECK(DropAllCapabilitiesOnCurrentThread());
+#endif
   return 0;
 }
 
