diff --git a/sandbox/linux/services/credentials.cc b/sandbox/linux/services/credentials.cc
index 76038af..3569070 100644
--- a/sandbox/linux/services/credentials.cc
+++ b/sandbox/linux/services/credentials.cc
@@ -327,9 +327,6 @@ pid_t Credentials::ForkAndDropCapabilitiesInChild() {
   if (pid != 0) {
     return pid;
   }
-
-  // Since we just forked, we are single threaded.
-  PCHECK(DropAllCapabilitiesOnCurrentThread());
   return 0;
 }
 
