diff --git a/sandbox/linux/services/credentials.cc b/sandbox/linux/services/credentials.cc
index 803af15..0b14177 100644
--- a/sandbox/linux/services/credentials.cc
+++ b/sandbox/linux/services/credentials.cc
@@ -331,8 +331,10 @@ pid_t Credentials::ForkAndDropCapabilitiesInChild() {
     return pid;
   }
 
+#if 0
   // Since we just forked, we are single threaded.
   PCHECK(DropAllCapabilitiesOnCurrentThread());
+#endif
   return 0;
 }
 
