diff --git a/sandbox/linux/services/credentials.cc b/sandbox/linux/services/credentials.cc
index fb6b6b4..b778f70 100644
--- a/sandbox/linux/services/credentials.cc
+++ b/sandbox/linux/services/credentials.cc
@@ -326,9 +326,6 @@ pid_t Credentials::ForkAndDropCapabilitiesInChild() {
   if (pid != 0) {
     return pid;
   }
-
-  // Since we just forked, we are single threaded.
-  PCHECK(DropAllCapabilitiesOnCurrentThread());
   return 0;
 }
 
