Index: ui/gfx/win/dpi.cc
===================================================================
--- ui/gfx/win/dpi.cc	(revision 274914)
+++ ui/gfx/win/dpi.cc	(working copy)
@@ -191,12 +191,8 @@
 }
 
 Point ScreenToDIPPoint(const Point& pixel_point) {
-  static float scaling_factor =
-      GetDeviceScaleFactor() > GetUnforcedDeviceScaleFactor() ?
-      1.0f / GetDeviceScaleFactor() :
-      1.0f;
   return ToFlooredPoint(ScalePoint(pixel_point,
-      scaling_factor));
+      1.0f / GetDeviceScaleFactor()));
 }
 
 Point DIPToScreenPoint(const Point& dip_point) {
