diff --git a/content/browser/web_contents/web_contents_impl.cc b/content/browser/web_contents/web_contents_impl.cc
index 1383981..d75f559 100644
--- a/content/browser/web_contents/web_contents_impl.cc
+++ b/content/browser/web_contents/web_contents_impl.cc
@@ -4119,6 +4119,7 @@ void WebContentsImpl::RendererUnresponsive(RenderViewHost* render_view_host) {
 
   if (rfhi->is_waiting_for_beforeunload_ack() ||
       rfhi->IsWaitingForUnloadACK()) {
+#if 0
     // Hang occurred while firing the beforeunload/unload handler.
     // Pretend the handler fired so tab closing continues as if it had.
     rvhi->set_sudden_termination_allowed(true);
@@ -4138,6 +4139,7 @@ void WebContentsImpl::RendererUnresponsive(RenderViewHost* render_view_host) {
     }
     if (close)
       Close(rvhi);
+#endif
     return;
   }
 
