diff --git a/content/browser/web_contents/web_contents_impl.cc b/content/browser/web_contents/web_contents_impl.cc
index 5621f82..68ca8c6 100644
--- a/content/browser/web_contents/web_contents_impl.cc
+++ b/content/browser/web_contents/web_contents_impl.cc
@@ -4561,6 +4561,7 @@ void WebContentsImpl::RendererUnresponsive(
 
   if (rfhi->is_waiting_for_beforeunload_ack() ||
       rfhi->IsWaitingForUnloadACK()) {
+#if 0
     // Hang occurred while firing the beforeunload/unload handler.
     // Pretend the handler fired so tab closing continues as if it had.
     GetRenderViewHost()->set_sudden_termination_allowed(true);
@@ -4580,6 +4581,7 @@ void WebContentsImpl::RendererUnresponsive(
     }
     if (close)
       Close();
+#endif
     return;
   }
 
