diff --git a/Source/platform/audio/AudioDSPKernel.cpp b/Source/platform/audio/AudioDSPKernel.cpp
index 719c44b..fdf26d6 100644
--- a/Source/platform/audio/AudioDSPKernel.cpp
+++ b/Source/platform/audio/AudioDSPKernel.cpp
@@ -31,8 +31,16 @@
 #include "config.h"
 #include "platform/audio/AudioDSPKernel.h"
 
+#include "platform/audio/AudioDSPKernelProcessor.h"
+
 namespace blink {
 
+AudioDSPKernel::AudioDSPKernel(AudioDSPKernelProcessor* kernelProcessor)
+    : m_kernelProcessor(kernelProcessor)
+    , m_sampleRate(kernelProcessor->sampleRate())
+{
+}
+
 AudioDSPKernel::~AudioDSPKernel()
 {
 }
diff --git a/Source/platform/audio/AudioDSPKernel.h b/Source/platform/audio/AudioDSPKernel.h
index a006864..05643b5 100644
--- a/Source/platform/audio/AudioDSPKernel.h
+++ b/Source/platform/audio/AudioDSPKernel.h
@@ -31,19 +31,17 @@
 #ifndef AudioDSPKernel_h
 #define AudioDSPKernel_h
 
-#include "platform/audio/AudioDSPKernelProcessor.h"
+#include "platform/PlatformExport.h"
 
 namespace blink {
 
+class AudioDSPKernelProcessor;
+
 // AudioDSPKernel does the processing for one channel of an AudioDSPKernelProcessor.
 
 class PLATFORM_EXPORT AudioDSPKernel {
 public:
-    AudioDSPKernel(AudioDSPKernelProcessor* kernelProcessor)
-        : m_kernelProcessor(kernelProcessor)
-        , m_sampleRate(kernelProcessor->sampleRate())
-    {
-    }
+    AudioDSPKernel(AudioDSPKernelProcessor* kernelProcessor);
 
     AudioDSPKernel(float sampleRate)
         : m_kernelProcessor(0)
diff --git a/Source/platform/audio/AudioDSPKernelProcessor.h b/Source/platform/audio/AudioDSPKernelProcessor.h
index 6a62da1..11916f2 100644
--- a/Source/platform/audio/AudioDSPKernelProcessor.h
+++ b/Source/platform/audio/AudioDSPKernelProcessor.h
@@ -32,6 +32,7 @@
 #define AudioDSPKernelProcessor_h
 
 #include "platform/audio/AudioBus.h"
+#include "platform/audio/AudioDSPKernel.h"
 #include "platform/audio/AudioProcessor.h"
 #include "wtf/OwnPtr.h"
 #include "wtf/PassOwnPtr.h"
