diff --git a/public/platform/WebDragData.h b/public/platform/WebDragData.h
index da9873b..c2beaa2 100644
--- a/public/platform/WebDragData.h
+++ b/public/platform/WebDragData.h
@@ -107,7 +107,7 @@ public:
     BLINK_EXPORT WebString filesystemId() const;
     BLINK_EXPORT void setFilesystemId(const WebString&);
 
-#if BLINK_IMPLEMENTATION
+#if INSIDE_BLINK
     explicit WebDragData(const PassRefPtrWillBeRawPtr<DataObject>&);
     WebDragData& operator=(const PassRefPtrWillBeRawPtr<DataObject>&);
     DataObject* getValue() const;
diff --git a/public/platform/WebGamepad.h b/public/platform/WebGamepad.h
index 63d63f1..8844654 100644
--- a/public/platform/WebGamepad.h
+++ b/public/platform/WebGamepad.h
@@ -26,7 +26,7 @@
 
 #include "WebCommon.h"
 
-#if BLINK_IMPLEMENTATION
+#if INSIDE_BLINK
 #include "wtf/Assertions.h"
 #endif
 
@@ -96,7 +96,7 @@ public:
     WebUChar mapping[mappingLengthCap];
 };
 
-#if BLINK_IMPLEMENTATION
+#if INSIDE_BLINK
 COMPILE_ASSERT(sizeof(WebGamepad) == 721, WebGamepad_has_wrong_size);
 #endif
 
diff --git a/public/platform/WebGamepads.h b/public/platform/WebGamepads.h
index 1fa1e81..dfedaf3 100644
--- a/public/platform/WebGamepads.h
+++ b/public/platform/WebGamepads.h
@@ -26,7 +26,7 @@
 
 #include "WebGamepad.h"
 
-#if BLINK_IMPLEMENTATION
+#if INSIDE_BLINK
 #include "wtf/Assertions.h"
 #endif
 
@@ -51,7 +51,7 @@ public:
     WebGamepad items[itemsLengthCap];
 };
 
-#if BLINK_IMPLEMENTATION
+#if INSIDE_BLINK
 COMPILE_ASSERT(sizeof(WebGamepads) == 2888, WebGamepads_has_wrong_size);
 #endif
 
diff --git a/public/platform/WebIDBDatabaseError.h b/public/platform/WebIDBDatabaseError.h
index d106d19..80911a0 100644
--- a/public/platform/WebIDBDatabaseError.h
+++ b/public/platform/WebIDBDatabaseError.h
@@ -54,7 +54,7 @@ public:
     BLINK_EXPORT void assign(const WebIDBDatabaseError&);
     BLINK_EXPORT void reset();
 
-#if BLINK_IMPLEMENTATION
+#if INSIDE_BLINK
     operator PassRefPtrWillBeRawPtr<DOMError>() const;
 #endif
 
diff --git a/public/platform/WebIDBKey.h b/public/platform/WebIDBKey.h
index 2aba653..5678d54 100644
--- a/public/platform/WebIDBKey.h
+++ b/public/platform/WebIDBKey.h
@@ -76,7 +76,7 @@ public:
     BLINK_EXPORT double date() const; // Only valid for DateType.
     BLINK_EXPORT double number() const; // Only valid for NumberType.
 
-#if BLINK_IMPLEMENTATION
+#if INSIDE_BLINK
     WebIDBKey(IDBKey*);
     WebIDBKey& operator=(IDBKey*);
     operator IDBKey*() const;
diff --git a/public/platform/WebIDBKeyPath.h b/public/platform/WebIDBKeyPath.h
index e15c6f2..ac68730 100644
--- a/public/platform/WebIDBKeyPath.h
+++ b/public/platform/WebIDBKeyPath.h
@@ -58,7 +58,7 @@ public:
     BLINK_EXPORT WebVector<WebString> array() const; // Only valid for ArrayType.
     BLINK_EXPORT WebString string() const; // Only valid for StringType.
 
-#if BLINK_IMPLEMENTATION
+#if INSIDE_BLINK
     WebIDBKeyPath(const IDBKeyPath&);
     WebIDBKeyPath& operator=(const IDBKeyPath&);
     operator const IDBKeyPath&() const;
diff --git a/public/platform/WebIDBKeyRange.h b/public/platform/WebIDBKeyRange.h
index 69fc5f6..39632d1 100644
--- a/public/platform/WebIDBKeyRange.h
+++ b/public/platform/WebIDBKeyRange.h
@@ -51,7 +51,7 @@ public:
     BLINK_EXPORT void assign(const WebIDBKey& lower, const WebIDBKey& upper, bool lowerOpen, bool upperOpen);
     BLINK_EXPORT void reset();
 
-#if BLINK_IMPLEMENTATION
+#if INSIDE_BLINK
     WebIDBKeyRange(IDBKeyRange*);
     WebIDBKeyRange& operator=(IDBKeyRange*);
     operator IDBKeyRange*() const;
diff --git a/public/platform/WebIDBMetadata.h b/public/platform/WebIDBMetadata.h
index 5f2e913..8c21f6b 100644
--- a/public/platform/WebIDBMetadata.h
+++ b/public/platform/WebIDBMetadata.h
@@ -78,7 +78,7 @@ struct WebIDBMetadata {
             , multiEntry(false) { }
     };
 
-#if BLINK_IMPLEMENTATION
+#if INSIDE_BLINK
     WebIDBMetadata(const IDBDatabaseMetadata&);
     operator IDBDatabaseMetadata() const;
 #endif
diff --git a/public/platform/WebRTCSessionDescription.h b/public/platform/WebRTCSessionDescription.h
index a4b1746..faaccf2 100644
--- a/public/platform/WebRTCSessionDescription.h
+++ b/public/platform/WebRTCSessionDescription.h
@@ -75,7 +75,7 @@ public:
     BLINK_PLATFORM_EXPORT WebString sdp() const;
     BLINK_PLATFORM_EXPORT void setSDP(const WebString&);
 
-#if BLINK_IMPLEMENTATION
+#if INSIDE_BLINK
     WebRTCSessionDescription(WebString type, WebString sdp)
     {
         this->initialize(type, sdp);
diff --git a/public/platform/WebScrollbarThemePainter.h b/public/platform/WebScrollbarThemePainter.h
index 48d1169..2b93768 100644
--- a/public/platform/WebScrollbarThemePainter.h
+++ b/public/platform/WebScrollbarThemePainter.h
@@ -59,7 +59,7 @@ public:
     BLINK_EXPORT void paintTickmarks(WebCanvas*, const WebRect&);
     BLINK_EXPORT void paintThumb(WebCanvas*, const WebRect&);
 
-#if BLINK_IMPLEMENTATION
+#if INSIDE_BLINK
     WebScrollbarThemePainter(ScrollbarTheme*, Scrollbar*);
 #endif
 
diff --git a/public/platform/WebWorkerRunLoop.h b/public/platform/WebWorkerRunLoop.h
index 748d111..e53442f 100644
--- a/public/platform/WebWorkerRunLoop.h
+++ b/public/platform/WebWorkerRunLoop.h
@@ -44,7 +44,7 @@ public:
     BLINK_EXPORT bool equals(const WebWorkerRunLoop&) const;
     BLINK_EXPORT bool lessThan(const WebWorkerRunLoop&) const;
 
-#if BLINK_IMPLEMENTATION
+#if INSIDE_BLINK
     WebWorkerRunLoop(WorkerThread*);
 #endif
 
diff --git a/public/web/WebAXObject.h b/public/web/WebAXObject.h
index 4d96c8f..5d64287 100644
--- a/public/web/WebAXObject.h
+++ b/public/web/WebAXObject.h
@@ -36,7 +36,7 @@
 #include "../platform/WebVector.h"
 #include "WebAXEnums.h"
 
-#if BLINK_IMPLEMENTATION
+#if INSIDE_BLINK
 namespace WTF { template <typename T> class PassRefPtr; }
 #endif
 
@@ -208,7 +208,7 @@ public:
     // Scroll this object to a given point in global coordinates of the top-level window.
     BLINK_EXPORT void scrollToGlobalPoint(const WebPoint&) const;
 
-#if BLINK_IMPLEMENTATION
+#if INSIDE_BLINK
     WebAXObject(const WTF::PassRefPtr<AXObject>&);
     WebAXObject& operator=(const WTF::PassRefPtr<AXObject>&);
     operator WTF::PassRefPtr<AXObject>() const;
diff --git a/public/web/WebArrayBufferView.h b/public/web/WebArrayBufferView.h
index cc02d7e..7caaeb5 100644
--- a/public/web/WebArrayBufferView.h
+++ b/public/web/WebArrayBufferView.h
@@ -57,7 +57,7 @@ public:
 
     BLINK_EXPORT static WebArrayBufferView* createFromV8Value(v8::Handle<v8::Value>);
 
-#if BLINK_IMPLEMENTATION
+#if INSIDE_BLINK
     WebArrayBufferView(const WTF::PassRefPtr<WTF::ArrayBufferView>&);
     WebArrayBufferView& operator=(const WTF::PassRefPtr<WTF::ArrayBufferView>&);
     operator WTF::PassRefPtr<WTF::ArrayBufferView>() const;
diff --git a/public/web/WebBlob.h b/public/web/WebBlob.h
index 1439ced..97c4818 100644
--- a/public/web/WebBlob.h
+++ b/public/web/WebBlob.h
@@ -37,7 +37,7 @@
 #include "public/platform/WebString.h"
 #include "public/platform/WebURL.h"
 
-#if BLINK_IMPLEMENTATION
+#if INSIDE_BLINK
 #include "platform/heap/Handle.h"
 #endif
 
@@ -76,7 +76,7 @@ public:
 
     BLINK_EXPORT v8::Handle<v8::Value>  toV8Value(v8::Handle<v8::Object> creationContext, v8::Isolate*);
 
-#if BLINK_IMPLEMENTATION
+#if INSIDE_BLINK
     explicit WebBlob(const PassRefPtrWillBeRawPtr<Blob>&);
     WebBlob& operator=(const PassRefPtrWillBeRawPtr<Blob>&);
 #endif
diff --git a/public/web/WebCachedURLRequest.h b/public/web/WebCachedURLRequest.h
index 89cb482..b91173d 100644
--- a/public/web/WebCachedURLRequest.h
+++ b/public/web/WebCachedURLRequest.h
@@ -52,7 +52,7 @@ public:
     BLINK_EXPORT bool forPreload() const;
     BLINK_EXPORT WebString initiatorName() const;
 
-#if BLINK_IMPLEMENTATION
+#if INSIDE_BLINK
     explicit WebCachedURLRequest(FetchRequest*);
 #endif
 
diff --git a/public/web/WebColorSuggestion.h b/public/web/WebColorSuggestion.h
index e413293..c206497 100644
--- a/public/web/WebColorSuggestion.h
+++ b/public/web/WebColorSuggestion.h
@@ -42,7 +42,7 @@ struct WebColorSuggestion {
     WebColor color;
     WebString label;
 
-#if BLINK_IMPLEMENTATION
+#if INSIDE_BLINK
     WebColorSuggestion(const ColorSuggestion&);
     WebColorSuggestion& operator=(const ColorSuggestion&);
 #endif
diff --git a/public/web/WebDOMError.h b/public/web/WebDOMError.h
index 869a542..6306d08 100644
--- a/public/web/WebDOMError.h
+++ b/public/web/WebDOMError.h
@@ -68,7 +68,7 @@ public:
 
     BLINK_EXPORT v8::Handle<v8::Value> toV8Value(v8::Handle<v8::Object> creationContext, v8::Isolate*);
 
-#if BLINK_IMPLEMENTATION
+#if INSIDE_BLINK
     explicit WebDOMError(const PassRefPtrWillBeRawPtr<DOMError>&);
     WebDOMError& operator=(const PassRefPtrWillBeRawPtr<DOMError>&);
 #endif
diff --git a/public/web/WebDOMEvent.h b/public/web/WebDOMEvent.h
index d10bded..9f3e84b 100644
--- a/public/web/WebDOMEvent.h
+++ b/public/web/WebDOMEvent.h
@@ -36,7 +36,7 @@
 #include "../platform/WebString.h"
 #include "WebNode.h"
 
-#if BLINK_IMPLEMENTATION
+#if INSIDE_BLINK
 namespace WTF { template <typename T> class PassRefPtr; }
 #endif
 
@@ -92,7 +92,7 @@ public:
     BLINK_EXPORT bool isProgressEvent() const;
     BLINK_EXPORT bool isXMLHttpRequestProgressEvent() const;
 
-#if BLINK_IMPLEMENTATION
+#if INSIDE_BLINK
     WebDOMEvent(const PassRefPtrWillBeRawPtr<Event>&);
     operator PassRefPtrWillBeRawPtr<Event>() const;
 #endif
@@ -112,7 +112,7 @@ public:
     }
 
 protected:
-#if BLINK_IMPLEMENTATION
+#if INSIDE_BLINK
     void assign(const PassRefPtrWillBeRawPtr<Event>&);
 
     template<typename T> T* unwrap()
diff --git a/public/web/WebDOMFileSystem.h b/public/web/WebDOMFileSystem.h
index 34520fd..5bb7e28 100644
--- a/public/web/WebDOMFileSystem.h
+++ b/public/web/WebDOMFileSystem.h
@@ -37,7 +37,7 @@
 #include "../platform/WebURL.h"
 #include "WebFrame.h"
 
-#if BLINK_IMPLEMENTATION
+#if INSIDE_BLINK
 #include "platform/heap/Handle.h"
 #endif
 
@@ -101,7 +101,7 @@ public:
 
     bool isNull() const { return m_private.isNull(); }
 
-#if BLINK_IMPLEMENTATION
+#if INSIDE_BLINK
     WebDOMFileSystem(DOMFileSystem*);
     WebDOMFileSystem& operator=(DOMFileSystem*);
 #endif
diff --git a/public/web/WebDOMMediaStreamTrack.h b/public/web/WebDOMMediaStreamTrack.h
index b66f7fc..fdc33b6 100644
--- a/public/web/WebDOMMediaStreamTrack.h
+++ b/public/web/WebDOMMediaStreamTrack.h
@@ -34,7 +34,7 @@
 #include "public/platform/WebMediaStreamTrack.h"
 #include "public/platform/WebPrivatePtr.h"
 
-#if BLINK_IMPLEMENTATION
+#if INSIDE_BLINK
 #include "wtf/PassRefPtr.h"
 #endif
 
@@ -67,7 +67,7 @@ public:
     BLINK_EXPORT WebMediaStreamTrack component() const;
 
 private:
-#if BLINK_IMPLEMENTATION
+#if INSIDE_BLINK
     WebDOMMediaStreamTrack(PassRefPtrWillBeRawPtr<MediaStreamTrack>);
 #endif
 
diff --git a/public/web/WebDOMMessageEvent.h b/public/web/WebDOMMessageEvent.h
index 5d8dc08..a19ae5a 100644
--- a/public/web/WebDOMMessageEvent.h
+++ b/public/web/WebDOMMessageEvent.h
@@ -34,7 +34,7 @@
 #include "WebDOMEvent.h"
 #include "WebSerializedScriptValue.h"
 
-#if BLINK_IMPLEMENTATION
+#if INSIDE_BLINK
 #include "core/events/Event.h"
 #include "core/events/MessageEvent.h"
 #endif
@@ -54,7 +54,7 @@ public:
 
     BLINK_EXPORT WebMessagePortChannelArray releaseChannels();
 
-#if BLINK_IMPLEMENTATION
+#if INSIDE_BLINK
     explicit WebDOMMessageEvent(const PassRefPtrWillBeRawPtr<MessageEvent>& e) : WebDOMEvent(e) { }
 #endif
 };
diff --git a/public/web/WebDateTimeSuggestion.h b/public/web/WebDateTimeSuggestion.h
index e2266b7..48d9f49 100644
--- a/public/web/WebDateTimeSuggestion.h
+++ b/public/web/WebDateTimeSuggestion.h
@@ -39,7 +39,7 @@ struct WebDateTimeSuggestion {
 
     WebDateTimeSuggestion() { }
 
-#if BLINK_IMPLEMENTATION
+#if INSIDE_BLINK
     WebDateTimeSuggestion(const DateTimeSuggestion&);
     WebDateTimeSuggestion& operator=(const DateTimeSuggestion&);
 #endif
diff --git a/public/web/WebDocument.h b/public/web/WebDocument.h
index 89f5549..6cdbf05 100644
--- a/public/web/WebDocument.h
+++ b/public/web/WebDocument.h
@@ -39,7 +39,7 @@
 #include "WebNode.h"
 #include "WebSecurityOrigin.h"
 
-#if BLINK_IMPLEMENTATION
+#if INSIDE_BLINK
 namespace WTF { template <typename T> class PassRefPtr; }
 #endif
 
@@ -145,7 +145,7 @@ public:
 
     BLINK_EXPORT WebURL manifestURL() const;
 
-#if BLINK_IMPLEMENTATION
+#if INSIDE_BLINK
     WebDocument(const PassRefPtrWillBeRawPtr<Document>&);
     WebDocument& operator=(const PassRefPtrWillBeRawPtr<Document>&);
     operator PassRefPtrWillBeRawPtr<Document>() const;
diff --git a/public/web/WebDocumentType.h b/public/web/WebDocumentType.h
index e3f33f5..b477b2b 100644
--- a/public/web/WebDocumentType.h
+++ b/public/web/WebDocumentType.h
@@ -33,7 +33,7 @@
 
 #include "WebNode.h"
 
-#if BLINK_IMPLEMENTATION
+#if INSIDE_BLINK
 namespace WTF { template <typename T> class PassRefPtr; }
 #endif
 
@@ -57,7 +57,7 @@ public:
 
     BLINK_EXPORT WebString name() const;
 
-#if BLINK_IMPLEMENTATION
+#if INSIDE_BLINK
     WebDocumentType(const PassRefPtrWillBeRawPtr<DocumentType>&);
     WebDocumentType& operator=(const PassRefPtrWillBeRawPtr<DocumentType>&);
     operator PassRefPtrWillBeRawPtr<DocumentType>() const;
diff --git a/public/web/WebElement.h b/public/web/WebElement.h
index 5db443c..652822c 100644
--- a/public/web/WebElement.h
+++ b/public/web/WebElement.h
@@ -82,7 +82,7 @@ public:
     // if there isn't any.
     BLINK_EXPORT WebImage imageContents();
 
-#if BLINK_IMPLEMENTATION
+#if INSIDE_BLINK
     WebElement(const PassRefPtrWillBeRawPtr<Element>&);
     WebElement& operator=(const PassRefPtrWillBeRawPtr<Element>&);
     operator PassRefPtrWillBeRawPtr<Element>() const;
diff --git a/public/web/WebElementCollection.h b/public/web/WebElementCollection.h
index d6bbba4..c962cc2 100644
--- a/public/web/WebElementCollection.h
+++ b/public/web/WebElementCollection.h
@@ -35,7 +35,7 @@
 #include "../platform/WebCommon.h"
 #include "../platform/WebPrivatePtr.h"
 
-#if BLINK_IMPLEMENTATION
+#if INSIDE_BLINK
 #include "platform/heap/Handle.h"
 namespace WTF { template <typename T> class PassRefPtr; }
 #endif
@@ -67,7 +67,7 @@ public:
     BLINK_EXPORT WebElement nextItem() const;
     BLINK_EXPORT WebElement firstItem() const;
 
-#if BLINK_IMPLEMENTATION
+#if INSIDE_BLINK
     WebElementCollection(const PassRefPtrWillBeRawPtr<HTMLCollection>&);
     WebElementCollection& operator=(const PassRefPtrWillBeRawPtr<HTMLCollection>&);
 #endif
diff --git a/public/web/WebFontDescription.h b/public/web/WebFontDescription.h
index cf0772f..130d71f 100644
--- a/public/web/WebFontDescription.h
+++ b/public/web/WebFontDescription.h
@@ -92,7 +92,7 @@ struct WebFontDescription {
     short letterSpacing;
     short wordSpacing;
 
-#if BLINK_IMPLEMENTATION
+#if INSIDE_BLINK
     WebFontDescription(const FontDescription&);
     operator FontDescription() const;
 #endif
diff --git a/public/web/WebFormControlElement.h b/public/web/WebFormControlElement.h
index 63de563..9ea8dfc 100644
--- a/public/web/WebFormControlElement.h
+++ b/public/web/WebFormControlElement.h
@@ -113,7 +113,7 @@ public:
 
     BLINK_EXPORT WebFormElement form() const;
 
-#if BLINK_IMPLEMENTATION
+#if INSIDE_BLINK
     WebFormControlElement(const PassRefPtrWillBeRawPtr<HTMLFormControlElement>&);
     WebFormControlElement& operator=(const PassRefPtrWillBeRawPtr<HTMLFormControlElement>&);
     operator PassRefPtrWillBeRawPtr<HTMLFormControlElement>() const;
diff --git a/public/web/WebFormElement.h b/public/web/WebFormElement.h
index cb2f5d0..f530ae9 100644
--- a/public/web/WebFormElement.h
+++ b/public/web/WebFormElement.h
@@ -78,7 +78,7 @@ public:
     };
     BLINK_EXPORT void finishRequestAutocomplete(WebFormElement::AutocompleteResult);
 
-#if BLINK_IMPLEMENTATION
+#if INSIDE_BLINK
     WebFormElement(const PassRefPtrWillBeRawPtr<HTMLFormElement>&);
     WebFormElement& operator=(const PassRefPtrWillBeRawPtr<HTMLFormElement>&);
     operator PassRefPtrWillBeRawPtr<HTMLFormElement>() const;
diff --git a/public/web/WebFrame.h b/public/web/WebFrame.h
index 92eea56..091a474 100644
--- a/public/web/WebFrame.h
+++ b/public/web/WebFrame.h
@@ -659,7 +659,7 @@ public:
     // text form. This is used only by layout tests.
     virtual WebString layerTreeAsText(bool showDebugInfo = false) const = 0;
 
-#if BLINK_IMPLEMENTATION
+#if INSIDE_BLINK
     static WebFrame* fromFrame(Frame*);
     static void traceChildren(Visitor*, WebFrame*);
 #endif
@@ -681,7 +681,7 @@ private:
     WebPrivateOwnPtr<OpenedFrameTracker> m_openedFrameTracker;
 };
 
-#if BLINK_IMPLEMENTATION
+#if INSIDE_BLINK
 Frame* toCoreFrame(const WebFrame*);
 #endif
 
diff --git a/public/web/WebGeolocationController.h b/public/web/WebGeolocationController.h
index 13e3757..21280e9 100644
--- a/public/web/WebGeolocationController.h
+++ b/public/web/WebGeolocationController.h
@@ -40,7 +40,7 @@ public:
     BLINK_EXPORT void positionChanged(const WebGeolocationPosition&);
     BLINK_EXPORT void errorOccurred(const WebGeolocationError&);
 
-#if BLINK_IMPLEMENTATION
+#if INSIDE_BLINK
     WebGeolocationController(GeolocationController* c)
         : m_private(c)
     {
diff --git a/public/web/WebGeolocationError.h b/public/web/WebGeolocationError.h
index 537432b..4ce46d3 100644
--- a/public/web/WebGeolocationError.h
+++ b/public/web/WebGeolocationError.h
@@ -49,7 +49,7 @@ public:
     BLINK_EXPORT void assign(const WebGeolocationError&);
     BLINK_EXPORT void reset();
 
-#if BLINK_IMPLEMENTATION
+#if INSIDE_BLINK
     WebGeolocationError(GeolocationError*);
     WebGeolocationError& operator=(GeolocationError*);
     operator GeolocationError*() const;
diff --git a/public/web/WebGeolocationPermissionRequest.h b/public/web/WebGeolocationPermissionRequest.h
index c9fe120..452c9fc 100644
--- a/public/web/WebGeolocationPermissionRequest.h
+++ b/public/web/WebGeolocationPermissionRequest.h
@@ -45,7 +45,7 @@ public:
     BLINK_EXPORT WebSecurityOrigin securityOrigin() const;
     BLINK_EXPORT void setIsAllowed(bool);
 
-#if BLINK_IMPLEMENTATION
+#if INSIDE_BLINK
     WebGeolocationPermissionRequest(Geolocation*);
     Geolocation* geolocation() const { return m_private.get(); }
 #endif
diff --git a/public/web/WebGeolocationPosition.h b/public/web/WebGeolocationPosition.h
index 6d7e03e..74eb73d 100644
--- a/public/web/WebGeolocationPosition.h
+++ b/public/web/WebGeolocationPosition.h
@@ -47,7 +47,7 @@ public:
     BLINK_EXPORT void assign(const WebGeolocationPosition&);
     BLINK_EXPORT void reset();
 
-#if BLINK_IMPLEMENTATION
+#if INSIDE_BLINK
     WebGeolocationPosition(GeolocationPosition*);
     WebGeolocationPosition& operator=(GeolocationPosition*);
     operator GeolocationPosition*() const;
diff --git a/public/web/WebHistoryItem.h b/public/web/WebHistoryItem.h
index 8029cbb..92bd287 100644
--- a/public/web/WebHistoryItem.h
+++ b/public/web/WebHistoryItem.h
@@ -115,7 +115,7 @@ public:
 
     BLINK_EXPORT WebVector<WebString> getReferencedFilePaths() const;
 
-#if BLINK_IMPLEMENTATION
+#if INSIDE_BLINK
     WebHistoryItem(const WTF::PassRefPtr<HistoryItem>&);
     WebHistoryItem& operator=(const WTF::PassRefPtr<HistoryItem>&);
     operator WTF::PassRefPtr<HistoryItem>() const;
diff --git a/public/web/WebHitTestResult.h b/public/web/WebHitTestResult.h
index 78a4cc4..63dc507 100644
--- a/public/web/WebHitTestResult.h
+++ b/public/web/WebHitTestResult.h
@@ -69,7 +69,7 @@ public:
     // Return whether an editable input element was hit.
     BLINK_EXPORT bool isContentEditable() const;
 
-#if BLINK_IMPLEMENTATION
+#if INSIDE_BLINK
     WebHitTestResult(const HitTestResult&);
     WebHitTestResult& operator=(const HitTestResult&);
 #endif
diff --git a/public/web/WebIconURL.h b/public/web/WebIconURL.h
index 8b333a22..c3c31f5 100644
--- a/public/web/WebIconURL.h
+++ b/public/web/WebIconURL.h
@@ -31,7 +31,7 @@
 #ifndef WebIconURL_h
 #define WebIconURL_h
 
-#if BLINK_IMPLEMENTATION
+#if INSIDE_BLINK
 #include "core/dom/IconURL.h"
 #endif
 #include "public/platform/WebSize.h"
@@ -75,7 +75,7 @@ public:
         return m_sizes;
     }
 
-#if BLINK_IMPLEMENTATION
+#if INSIDE_BLINK
     WebIconURL(const IconURL& iconURL)
         : m_iconType(static_cast<Type>(iconURL.m_iconType))
         , m_iconURL(iconURL.m_iconURL)
diff --git a/public/web/WebInputElement.h b/public/web/WebInputElement.h
index de38d60..1f63a9e 100644
--- a/public/web/WebInputElement.h
+++ b/public/web/WebInputElement.h
@@ -84,7 +84,7 @@ public:
     // If true, forces the text of the element to be visible.
     BLINK_EXPORT void setShouldRevealPassword(bool value);
 
-#if BLINK_IMPLEMENTATION
+#if INSIDE_BLINK
     WebInputElement(const PassRefPtrWillBeRawPtr<HTMLInputElement>&);
     WebInputElement& operator=(const PassRefPtrWillBeRawPtr<HTMLInputElement>&);
     operator PassRefPtrWillBeRawPtr<HTMLInputElement>() const;
diff --git a/public/web/WebLabelElement.h b/public/web/WebLabelElement.h
index 69c8e93..a1a8c8a 100644
--- a/public/web/WebLabelElement.h
+++ b/public/web/WebLabelElement.h
@@ -33,7 +33,7 @@
 
 #include "WebElement.h"
 
-#if BLINK_IMPLEMENTATION
+#if INSIDE_BLINK
 namespace WTF { template <typename T> class PassRefPtr; }
 #endif
 
@@ -57,7 +57,7 @@ public:
 
     BLINK_EXPORT WebElement correspondingControl();
 
-#if BLINK_IMPLEMENTATION
+#if INSIDE_BLINK
     WebLabelElement(const PassRefPtrWillBeRawPtr<HTMLLabelElement>&);
     WebLabelElement& operator=(const PassRefPtrWillBeRawPtr<HTMLLabelElement>&);
     operator PassRefPtrWillBeRawPtr<HTMLLabelElement>() const;
diff --git a/public/web/WebMIDIPermissionRequest.h b/public/web/WebMIDIPermissionRequest.h
index 87cafd6..1e9b1b0 100644
--- a/public/web/WebMIDIPermissionRequest.h
+++ b/public/web/WebMIDIPermissionRequest.h
@@ -52,7 +52,7 @@ public:
 
     BLINK_EXPORT bool equals(const WebMIDIPermissionRequest&) const;
 
-#if BLINK_IMPLEMENTATION
+#if INSIDE_BLINK
     explicit WebMIDIPermissionRequest(MIDIAccessInitializer*);
 
     MIDIAccessInitializer* midiAccessInitializer() const { return m_initializer; }
diff --git a/public/web/WebMediaDevicesRequest.h b/public/web/WebMediaDevicesRequest.h
index a0126e7..dd6aef2 100644
--- a/public/web/WebMediaDevicesRequest.h
+++ b/public/web/WebMediaDevicesRequest.h
@@ -60,7 +60,7 @@ public:
 
     BLINK_EXPORT void requestSucceeded(WebVector<WebMediaDeviceInfo>);
 
-#if BLINK_IMPLEMENTATION
+#if INSIDE_BLINK
     WebMediaDevicesRequest(MediaDevicesRequest*);
     operator MediaDevicesRequest*() const;
 #endif
diff --git a/public/web/WebNode.h b/public/web/WebNode.h
index 1b3df2e..8c9f684 100644
--- a/public/web/WebNode.h
+++ b/public/web/WebNode.h
@@ -139,13 +139,13 @@ public:
         return res;
     }
 
-#if BLINK_IMPLEMENTATION
+#if INSIDE_BLINK
     WebNode(const PassRefPtrWillBeRawPtr<Node>&);
     WebNode& operator=(const PassRefPtrWillBeRawPtr<Node>&);
     operator PassRefPtrWillBeRawPtr<Node>() const;
 #endif
 
-#if BLINK_IMPLEMENTATION
+#if INSIDE_BLINK
     template<typename T> T* unwrap()
     {
         return static_cast<T*>(m_private.get());
diff --git a/public/web/WebNodeList.h b/public/web/WebNodeList.h
index 4e7dcf9..dad58c5 100644
--- a/public/web/WebNodeList.h
+++ b/public/web/WebNodeList.h
@@ -34,7 +34,7 @@
 #include "public/platform/WebCommon.h"
 #include "public/platform/WebPrivatePtr.h"
 
-#if BLINK_IMPLEMENTATION
+#if INSIDE_BLINK
 #include "platform/heap/Handle.h"
 namespace WTF { template <typename T> class PassRefPtr; }
 #endif
@@ -63,7 +63,7 @@ public:
     BLINK_EXPORT unsigned length() const;
     BLINK_EXPORT WebNode item(size_t) const;
 
-#if BLINK_IMPLEMENTATION
+#if INSIDE_BLINK
     WebNodeList(const PassRefPtrWillBeRawPtr<NodeList>&);
     WebNodeList& operator=(const PassRefPtrWillBeRawPtr<NodeList>&);
 #endif
diff --git a/public/web/WebNotification.h b/public/web/WebNotification.h
index 347b8ff..effcb8a 100644
--- a/public/web/WebNotification.h
+++ b/public/web/WebNotification.h
@@ -36,7 +36,7 @@
 #include "../platform/WebString.h"
 #include "WebTextDirection.h"
 
-#if BLINK_IMPLEMENTATION
+#if INSIDE_BLINK
 namespace WTF { template <typename T> class PassRefPtr; }
 #endif
 
@@ -91,7 +91,7 @@ public:
     WebString replaceId() const { return tag(); }
     void dispatchDisplayEvent() { dispatchShowEvent(); }
 
-#if BLINK_IMPLEMENTATION
+#if INSIDE_BLINK
     WebNotification(Notification*);
     WebNotification& operator=(Notification*);
 #endif
diff --git a/public/web/WebOptionElement.h b/public/web/WebOptionElement.h
index dfbba7e..97e89d5 100644
--- a/public/web/WebOptionElement.h
+++ b/public/web/WebOptionElement.h
@@ -59,7 +59,7 @@ public:
     BLINK_EXPORT WebString label() const;
     BLINK_EXPORT bool isEnabled() const;
 
-#if BLINK_IMPLEMENTATION
+#if INSIDE_BLINK
     WebOptionElement(const PassRefPtrWillBeRawPtr<HTMLOptionElement>&);
     WebOptionElement& operator=(const PassRefPtrWillBeRawPtr<HTMLOptionElement>&);
     operator PassRefPtrWillBeRawPtr<HTMLOptionElement>() const;
diff --git a/public/web/WebPerformance.h b/public/web/WebPerformance.h
index 6fd8c5b..1f7c24a 100644
--- a/public/web/WebPerformance.h
+++ b/public/web/WebPerformance.h
@@ -35,7 +35,7 @@
 #include "../platform/WebPrivatePtr.h"
 #include "WebNavigationType.h"
 
-#if BLINK_IMPLEMENTATION
+#if INSIDE_BLINK
 #include "platform/heap/Handle.h"
 #endif
 
@@ -85,7 +85,7 @@ public:
     BLINK_EXPORT double loadEventStart() const;
     BLINK_EXPORT double loadEventEnd() const;
 
-#if BLINK_IMPLEMENTATION
+#if INSIDE_BLINK
     WebPerformance(const PassRefPtrWillBeRawPtr<Performance>&);
     WebPerformance& operator=(const PassRefPtrWillBeRawPtr<Performance>&);
 #endif
diff --git a/public/web/WebPluginDocument.h b/public/web/WebPluginDocument.h
index 7424bae..09e63d9 100644
--- a/public/web/WebPluginDocument.h
+++ b/public/web/WebPluginDocument.h
@@ -53,7 +53,7 @@ public:
 
     BLINK_EXPORT WebPlugin* plugin();
 
-#if BLINK_IMPLEMENTATION
+#if INSIDE_BLINK
     WebPluginDocument(const PassRefPtrWillBeRawPtr<PluginDocument>&);
     WebPluginDocument& operator=(const PassRefPtrWillBeRawPtr<PluginDocument>&);
     operator PassRefPtrWillBeRawPtr<PluginDocument>() const;
diff --git a/public/web/WebRange.h b/public/web/WebRange.h
index 2a27890..ea79af2 100644
--- a/public/web/WebRange.h
+++ b/public/web/WebRange.h
@@ -72,7 +72,7 @@ public:
 
     BLINK_EXPORT static WebRange fromDocumentRange(WebLocalFrame*, int start, int length);
 
-#if BLINK_IMPLEMENTATION
+#if INSIDE_BLINK
     WebRange(const PassRefPtrWillBeRawPtr<Range>&);
     operator PassRefPtrWillBeRawPtr<Range>() const;
 #endif
diff --git a/public/web/WebSecurityOrigin.h b/public/web/WebSecurityOrigin.h
index 0a3474d..47e153e 100644
--- a/public/web/WebSecurityOrigin.h
+++ b/public/web/WebSecurityOrigin.h
@@ -33,7 +33,7 @@
 
 #include "../platform/WebCommon.h"
 
-#if BLINK_IMPLEMENTATION
+#if INSIDE_BLINK
 namespace WTF { template <typename T> class PassRefPtr; }
 #endif
 
@@ -106,7 +106,7 @@ public:
     // Allows this WebSecurityOrigin access to local resources.
     BLINK_EXPORT void grantLoadLocalResources() const;
 
-#if BLINK_IMPLEMENTATION
+#if INSIDE_BLINK
     WebSecurityOrigin(const WTF::PassRefPtr<SecurityOrigin>&);
     WebSecurityOrigin& operator=(const WTF::PassRefPtr<SecurityOrigin>&);
     operator WTF::PassRefPtr<SecurityOrigin>() const;
diff --git a/public/web/WebSelectElement.h b/public/web/WebSelectElement.h
index c60c802..936384a 100644
--- a/public/web/WebSelectElement.h
+++ b/public/web/WebSelectElement.h
@@ -54,7 +54,7 @@ public:
 
     BLINK_EXPORT WebVector<WebElement> listItems() const;
 
-#if BLINK_IMPLEMENTATION
+#if INSIDE_BLINK
     WebSelectElement(const PassRefPtrWillBeRawPtr<HTMLSelectElement>&);
     WebSelectElement& operator=(const PassRefPtrWillBeRawPtr<HTMLSelectElement>&);
     operator PassRefPtrWillBeRawPtr<HTMLSelectElement>() const;
diff --git a/public/web/WebSerializedScriptValue.h b/public/web/WebSerializedScriptValue.h
index ec1ba98..a47a471 100644
--- a/public/web/WebSerializedScriptValue.h
+++ b/public/web/WebSerializedScriptValue.h
@@ -75,7 +75,7 @@ public:
     // Convert the serialized value to a parsed v8 value.
     BLINK_EXPORT v8::Handle<v8::Value> deserialize();
 
-#if BLINK_IMPLEMENTATION
+#if INSIDE_BLINK
     WebSerializedScriptValue(const WTF::PassRefPtr<SerializedScriptValue>&);
     WebSerializedScriptValue& operator=(const WTF::PassRefPtr<SerializedScriptValue>&);
     operator WTF::PassRefPtr<SerializedScriptValue>() const;
diff --git a/public/web/WebSpeechGrammar.h b/public/web/WebSpeechGrammar.h
index 0d50505..3891418 100644
--- a/public/web/WebSpeechGrammar.h
+++ b/public/web/WebSpeechGrammar.h
@@ -46,7 +46,7 @@ public:
     BLINK_EXPORT void reset();
     BLINK_EXPORT void assign(const WebSpeechGrammar&);
 
-#if BLINK_IMPLEMENTATION
+#if INSIDE_BLINK
     explicit WebSpeechGrammar(SpeechGrammar*);
     WebSpeechGrammar& operator=(SpeechGrammar*);
 #endif
diff --git a/public/web/WebSpeechRecognitionHandle.h b/public/web/WebSpeechRecognitionHandle.h
index aaf284f..8f1e81b 100644
--- a/public/web/WebSpeechRecognitionHandle.h
+++ b/public/web/WebSpeechRecognitionHandle.h
@@ -58,7 +58,7 @@ public:
     BLINK_EXPORT bool equals(const WebSpeechRecognitionHandle&) const;
     BLINK_EXPORT bool lessThan(const WebSpeechRecognitionHandle&) const;
 
-#if BLINK_IMPLEMENTATION
+#if INSIDE_BLINK
     explicit WebSpeechRecognitionHandle(SpeechRecognition*);
     WebSpeechRecognitionHandle& operator=(SpeechRecognition*);
     operator SpeechRecognition*() const;
diff --git a/public/web/WebSpeechRecognitionResult.h b/public/web/WebSpeechRecognitionResult.h
index 2999562..bfa9ee1 100644
--- a/public/web/WebSpeechRecognitionResult.h
+++ b/public/web/WebSpeechRecognitionResult.h
@@ -45,7 +45,7 @@ public:
     BLINK_EXPORT void assign(const WebSpeechRecognitionResult&);
     BLINK_EXPORT void reset();
 
-#if BLINK_IMPLEMENTATION
+#if INSIDE_BLINK
     operator SpeechRecognitionResult*() const;
 #endif
 
diff --git a/public/web/WebTextAreaElement.h b/public/web/WebTextAreaElement.h
index 04b5e73..f7d8b7c 100644
--- a/public/web/WebTextAreaElement.h
+++ b/public/web/WebTextAreaElement.h
@@ -50,7 +50,7 @@ public:
     }
     void assign(const WebTextAreaElement& element) { WebFormControlElement::assign(element); }
 
-#if BLINK_IMPLEMENTATION
+#if INSIDE_BLINK
     WebTextAreaElement(const PassRefPtrWillBeRawPtr<HTMLTextAreaElement>&);
     WebTextAreaElement& operator=(const PassRefPtrWillBeRawPtr<HTMLTextAreaElement>&);
     operator PassRefPtrWillBeRawPtr<HTMLTextAreaElement>() const;
diff --git a/public/web/WebTextCheckingResult.h b/public/web/WebTextCheckingResult.h
index 4becea1..8b51662 100644
--- a/public/web/WebTextCheckingResult.h
+++ b/public/web/WebTextCheckingResult.h
@@ -58,7 +58,7 @@ struct WebTextCheckingResult {
     {
     }
 
-#if BLINK_IMPLEMENTATION
+#if INSIDE_BLINK
     operator TextCheckingResult() const;
 #endif
 
diff --git a/public/web/WebTextDirection.h b/public/web/WebTextDirection.h
index cd350c9..045ec6a 100644
--- a/public/web/WebTextDirection.h
+++ b/public/web/WebTextDirection.h
@@ -31,7 +31,7 @@
 #ifndef WebTextDirection_h
 #define WebTextDirection_h
 
-#if BLINK_IMPLEMENTATION
+#if INSIDE_BLINK
 #include "platform/text/TextDirection.h"
 #endif
 
@@ -45,7 +45,7 @@ enum WebTextDirection {
     WebTextDirectionLast = WebTextDirectionRightToLeft
 };
 
-#if BLINK_IMPLEMENTATION
+#if INSIDE_BLINK
 inline WebTextDirection toWebTextDirection(TextDirection direction)
 {
     switch (direction) {
diff --git a/public/web/WebTextRun.h b/public/web/WebTextRun.h
index c4a64e2..50c293f 100644
--- a/public/web/WebTextRun.h
+++ b/public/web/WebTextRun.h
@@ -54,7 +54,7 @@ struct WebTextRun {
     bool rtl;
     bool directionalOverride;
 
-#if BLINK_IMPLEMENTATION
+#if INSIDE_BLINK
     // The resulting blink::TextRun will refer to the text in this
     // struct, so "this" must outlive the WebCore text run.
     operator TextRun() const;
diff --git a/public/web/WebUserGestureToken.h b/public/web/WebUserGestureToken.h
index 1721724..1629560 100644
--- a/public/web/WebUserGestureToken.h
+++ b/public/web/WebUserGestureToken.h
@@ -56,7 +56,7 @@ public:
     BLINK_EXPORT void setJavascriptPrompt();
     bool isNull() const { return m_token.isNull(); }
 
-#if BLINK_IMPLEMENTATION
+#if INSIDE_BLINK
     explicit WebUserGestureToken(PassRefPtr<UserGestureToken>);
     operator PassRefPtr<UserGestureToken>() const;
 #endif
diff --git a/public/web/WebUserMediaRequest.h b/public/web/WebUserMediaRequest.h
index e202ae1..e7f04ed 100644
--- a/public/web/WebUserMediaRequest.h
+++ b/public/web/WebUserMediaRequest.h
@@ -79,7 +79,7 @@ public:
     // DEPRECATED
     BLINK_EXPORT void requestFailed(const WebString& description = WebString()) { requestDenied(description); }
 
-#if BLINK_IMPLEMENTATION
+#if INSIDE_BLINK
     WebUserMediaRequest(UserMediaRequest*);
     operator UserMediaRequest*() const;
 #endif
diff --git a/public/web/WebWindowFeatures.h b/public/web/WebWindowFeatures.h
index 814eeaa..cfc0027 100644
--- a/public/web/WebWindowFeatures.h
+++ b/public/web/WebWindowFeatures.h
@@ -35,7 +35,7 @@
 #include "../platform/WebString.h"
 #include "../platform/WebVector.h"
 
-#if BLINK_IMPLEMENTATION
+#if INSIDE_BLINK
 #include "core/page/WindowFeatures.h"
 #endif
 
@@ -83,7 +83,7 @@ struct WebWindowFeatures {
     }
 
 
-#if BLINK_IMPLEMENTATION
+#if INSIDE_BLINK
     WebWindowFeatures(const WindowFeatures& f)
         : x(f.x)
         , xSet(f.xSet)
