diff --git a/Source/core/loader/ProgressTracker.cpp b/Source/core/loader/ProgressTracker.cpp
index 251f4d1..c7263ba 100644
--- a/Source/core/loader/ProgressTracker.cpp
+++ b/Source/core/loader/ProgressTracker.cpp
@@ -93,7 +93,7 @@ void ProgressTracker::dispose()
 {
     if (m_frame->isLoading())
         progressCompleted();
-    ASSERT(!m_frame->isLoading());
+    // ASSERT(!m_frame->isLoading());
 }
 
 double ProgressTracker::estimatedProgress() const
