diff --git a/include/v8.h b/include/v8.h
index 277cbd442a..a99fdf9fbe 100644
--- a/include/v8.h
+++ b/include/v8.h
@@ -4266,6 +4266,13 @@ class V8_EXPORT ArrayBuffer : public Object {
      */
     virtual void* AllocateUninitialized(size_t length) = 0;

+    /**
+     * Free the memory block of size |length|, pointed to by |data|.
+     * That memory must be previously allocated by |Allocate| and not yet freed
+     * with a call to |Free| or |Realloc|
+     */
+    virtual void* Realloc(void* data, size_t length);
+
     /**
      * Reserved |length| bytes, but do not commit the memory. Must call
      * |SetProtection| to make memory accessible.
diff --git a/src/api.cc b/src/api.cc
index 8531cd5c05..1157d4eec2 100644
--- a/src/api.cc
+++ b/src/api.cc
@@ -458,6 +458,10 @@ void V8::SetSnapshotDataBlob(StartupData* snapshot_blob) {
 
 void* v8::ArrayBuffer::Allocator::Reserve(size_t length) { UNIMPLEMENTED(); }
 
+void* v8::ArrayBuffer::Allocator::Realloc(void* data, size_t length) {
+  UNIMPLEMENTED();
+}
+
 void v8::ArrayBuffer::Allocator::Free(void* data, size_t length,
                                       AllocationMode mode) {
   UNIMPLEMENTED();
