diff --git a/content/browser/device_monitor_mac.mm b/content/browser/device_monitor_mac.mm
index 36e9a57..87b8889 100644
--- a/content/browser/device_monitor_mac.mm
+++ b/content/browser/device_monitor_mac.mm
@@ -4,7 +4,9 @@
 
 #include "content/browser/device_monitor_mac.h"
 
+#ifndef MAS_BUILD
 #import <QTKit/QTKit.h>
+#endif
 
 #include <set>
 
@@ -150,6 +152,7 @@ class QTKitMonitorImpl : public DeviceMonitorMacImpl {
 
 QTKitMonitorImpl::QTKitMonitorImpl(content::DeviceMonitorMac* monitor)
     : DeviceMonitorMacImpl(monitor) {
+#ifndef MAS_BUILD
   NSNotificationCenter* nc = [NSNotificationCenter defaultCenter];
   device_arrival_ =
       [nc addObserverForName:QTCaptureDeviceWasConnectedNotification
@@ -169,25 +172,31 @@ QTKitMonitorImpl::QTKitMonitorImpl(content::DeviceMonitorMac* monitor)
                        queue:nil
                   usingBlock:^(NSNotification* notification) {
                       OnAttributeChanged(notification);}];
+#endif
 }
 
 QTKitMonitorImpl::~QTKitMonitorImpl() {
+#ifndef MAS_BUILD
   NSNotificationCenter* nc = [NSNotificationCenter defaultCenter];
   [nc removeObserver:device_arrival_];
   [nc removeObserver:device_removal_];
   [nc removeObserver:device_change_];
+#endif
 }
 
 void QTKitMonitorImpl::OnAttributeChanged(
     NSNotification* notification) {
+#ifndef MAS_BUILD
   if ([[[notification userInfo]
          objectForKey:QTCaptureDeviceChangedAttributeKey]
       isEqualToString:QTCaptureDeviceSuspendedAttribute]) {
     OnDeviceChanged();
   }
+#endif
 }
 
 void QTKitMonitorImpl::OnDeviceChanged() {
+#ifndef MAS_BUILD
   std::vector<DeviceInfo> snapshot_devices;
 
   NSArray* devices = [QTCaptureDevice inputDevices];
@@ -212,6 +221,7 @@ void QTKitMonitorImpl::OnDeviceChanged() {
         DeviceInfo([[device uniqueID] UTF8String], device_type));
   }
   ConsolidateDevicesListAndNotify(snapshot_devices);
+#endif
 }
 
 // Forward declaration for use by CrAVFoundationDeviceObserver.
diff --git a/media/capture/video/mac/video_capture_device_factory_mac.mm b/media/capture/video/mac/video_capture_device_factory_mac.mm
index 3d7c795..eb46b84 100644
--- a/media/capture/video/mac/video_capture_device_factory_mac.mm
+++ b/media/capture/video/mac/video_capture_device_factory_mac.mm
@@ -15,7 +15,9 @@
 #import "media/capture/video/mac/video_capture_device_avfoundation_mac.h"
 #import "media/capture/video/mac/video_capture_device_decklink_mac.h"
 #include "media/capture/video/mac/video_capture_device_mac.h"
+#ifndef MAS_BUILD
 #import "media/capture/video/mac/video_capture_device_qtkit_mac.h"
+#endif
 
 namespace media {
 
@@ -55,6 +57,7 @@ EnumerateDevicesUsingQTKit() {
 
   scoped_ptr<VideoCaptureDevice::Names> device_names(
       new VideoCaptureDevice::Names());
+#ifndef MAS_BUILD
   NSMutableDictionary* capture_devices =
       [[[NSMutableDictionary alloc] init] autorelease];
   [VideoCaptureDeviceQTKit getDeviceNames:capture_devices];
@@ -66,6 +69,7 @@ EnumerateDevicesUsingQTKit() {
       name.set_is_blacklisted(true);
     device_names->push_back(name);
   }
+#endif
   return device_names.Pass();
 }
 
diff --git a/media/capture/video/mac/video_capture_device_mac.mm b/media/capture/video/mac/video_capture_device_mac.mm
index 34c38f7..15a5658 100644
--- a/media/capture/video/mac/video_capture_device_mac.mm
+++ b/media/capture/video/mac/video_capture_device_mac.mm
@@ -21,7 +21,9 @@
 #include "media/base/timestamp_constants.h"
 #import "media/capture/video/mac/platform_video_capturing_mac.h"
 #import "media/capture/video/mac/video_capture_device_avfoundation_mac.h"
+#ifndef MAS_BUILD
 #import "media/capture/video/mac/video_capture_device_qtkit_mac.h"
+#endif
 #include "ui/gfx/geometry/size.h"
 
 @implementation DeviceNameAndTransportType
@@ -451,8 +453,10 @@ bool VideoCaptureDeviceMac::Init(
     capture_device_ =
         [[VideoCaptureDeviceAVFoundation alloc] initWithFrameReceiver:this];
   } else {
+#ifndef MAS_BUILD
     capture_device_ =
         [[VideoCaptureDeviceQTKit alloc] initWithFrameReceiver:this];
+#endif
   }
 
   if (!capture_device_)
diff --git a/media/media.gyp b/media/media.gyp
index b56a23b..5c4932c 100644
--- a/media/media.gyp
+++ b/media/media.gyp
@@ -461,8 +461,8 @@
         'capture/video/mac/video_capture_device_factory_mac.mm',
         'capture/video/mac/video_capture_device_mac.h',
         'capture/video/mac/video_capture_device_mac.mm',
-        'capture/video/mac/video_capture_device_qtkit_mac.h',
-        'capture/video/mac/video_capture_device_qtkit_mac.mm',
+        # 'capture/video/mac/video_capture_device_qtkit_mac.h',
+        # 'capture/video/mac/video_capture_device_qtkit_mac.mm',
         'capture/video/video_capture_device.cc',
         'capture/video/video_capture_device.h',
         'capture/video/video_capture_device_factory.cc',
@@ -990,7 +990,7 @@
               '$(SDKROOT)/System/Library/Frameworks/CoreAudio.framework',
               '$(SDKROOT)/System/Library/Frameworks/CoreVideo.framework',
               '$(SDKROOT)/System/Library/Frameworks/OpenGL.framework',
-              '$(SDKROOT)/System/Library/Frameworks/QTKit.framework',
+              # '$(SDKROOT)/System/Library/Frameworks/QTKit.framework',
             ],
           },
         }],
