diff --git a/media/capture/device_monitor_mac.mm b/media/capture/device_monitor_mac.mm
index aab9884..8951a44 100644
--- a/media/capture/device_monitor_mac.mm
+++ b/media/capture/device_monitor_mac.mm
@@ -4,7 +4,9 @@
 
 #include "media/capture/device_monitor_mac.h"
 
+#ifndef MAS_BUILD
 #import <QTKit/QTKit.h>
+#endif
 
 #include <set>
 
@@ -143,6 +145,7 @@ class QTKitMonitorImpl : public DeviceMonitorMacImpl {
 
 QTKitMonitorImpl::QTKitMonitorImpl(media::DeviceMonitorMac* monitor)
     : DeviceMonitorMacImpl(monitor) {
+#ifndef MAS_BUILD
   NSNotificationCenter* nc = [NSNotificationCenter defaultCenter];
   device_arrival_ =
       [nc addObserverForName:QTCaptureDeviceWasConnectedNotification
@@ -165,23 +168,29 @@ QTKitMonitorImpl::QTKitMonitorImpl(media::DeviceMonitorMac* monitor)
                   usingBlock:^(NSNotification* notification) {
                     OnAttributeChanged(notification);
                   }];
+#endif
 }
 
 QTKitMonitorImpl::~QTKitMonitorImpl() {
+#ifndef MAS_BUILD
   NSNotificationCenter* nc = [NSNotificationCenter defaultCenter];
   [nc removeObserver:device_arrival_];
   [nc removeObserver:device_removal_];
   [nc removeObserver:device_change_];
+#endif
 }
 
 void QTKitMonitorImpl::OnAttributeChanged(NSNotification* notification) {
+#ifndef MAS_BUILD
   if ([[[notification userInfo] objectForKey:QTCaptureDeviceChangedAttributeKey]
           isEqualToString:QTCaptureDeviceSuspendedAttribute]) {
     OnDeviceChanged();
   }
+#endif
 }
 
 void QTKitMonitorImpl::OnDeviceChanged() {
+#ifndef MAS_BUILD
   std::vector<DeviceInfo> snapshot_devices;
 
   NSArray* devices = [QTCaptureDevice inputDevices];
@@ -206,6 +215,7 @@ void QTKitMonitorImpl::OnDeviceChanged() {
         DeviceInfo([[device uniqueID] UTF8String], device_type));
   }
   ConsolidateDevicesListAndNotify(snapshot_devices);
+#endif
 }
 
 // Forward declaration for use by CrAVFoundationDeviceObserver.
diff --git a/media/capture/video/mac/video_capture_device_factory_mac.mm b/media/capture/video/mac/video_capture_device_factory_mac.mm
index 5d92f7e..09b4d13 100644
--- a/media/capture/video/mac/video_capture_device_factory_mac.mm
+++ b/media/capture/video/mac/video_capture_device_factory_mac.mm
@@ -19,7 +19,9 @@
 #import "media/capture/video/mac/video_capture_device_avfoundation_mac.h"
 #import "media/capture/video/mac/video_capture_device_decklink_mac.h"
 #include "media/capture/video/mac/video_capture_device_mac.h"
+#ifndef MAS_BUILD
 #import "media/capture/video/mac/video_capture_device_qtkit_mac.h"
+#endif
 
 namespace media {
 
@@ -59,6 +61,7 @@ EnumerateDevicesUsingQTKit() {
 
   scoped_ptr<VideoCaptureDevice::Names> device_names(
       new VideoCaptureDevice::Names());
+#ifndef MAS_BUILD
   NSMutableDictionary* capture_devices =
       [[[NSMutableDictionary alloc] init] autorelease];
   [VideoCaptureDeviceQTKit getDeviceNames:capture_devices];
@@ -70,6 +73,7 @@ EnumerateDevicesUsingQTKit() {
       name.set_is_blacklisted(true);
     device_names->push_back(name);
   }
+#endif
   return device_names;
 }
 
diff --git a/media/capture/video/mac/video_capture_device_mac.mm b/media/capture/video/mac/video_capture_device_mac.mm
index 432fbfb..e9932c5 100644
--- a/media/capture/video/mac/video_capture_device_mac.mm
+++ b/media/capture/video/mac/video_capture_device_mac.mm
@@ -27,7 +27,9 @@
 #include "media/base/timestamp_constants.h"
 #import "media/capture/video/mac/platform_video_capturing_mac.h"
 #import "media/capture/video/mac/video_capture_device_avfoundation_mac.h"
+#ifndef MAS_BUILD
 #import "media/capture/video/mac/video_capture_device_qtkit_mac.h"
+#endif
 #include "ui/gfx/geometry/size.h"
 
 @implementation DeviceNameAndTransportType
@@ -457,8 +459,10 @@ bool VideoCaptureDeviceMac::Init(
     capture_device_ =
         [[VideoCaptureDeviceAVFoundation alloc] initWithFrameReceiver:this];
   } else {
+#ifndef MAS_BUILD
     capture_device_ =
         [[VideoCaptureDeviceQTKit alloc] initWithFrameReceiver:this];
+#endif
   }
 
   if (!capture_device_)
diff --git a/media/media.gyp b/media/media.gyp
index 329f0c7..26c6597 100644
--- a/media/media.gyp
+++ b/media/media.gyp
@@ -956,7 +956,6 @@
               '$(SDKROOT)/System/Library/Frameworks/CoreAudio.framework',
               '$(SDKROOT)/System/Library/Frameworks/CoreVideo.framework',
               '$(SDKROOT)/System/Library/Frameworks/OpenGL.framework',
-              '$(SDKROOT)/System/Library/Frameworks/QTKit.framework',
             ],
           },
         }],
