diff --git a/src/base/logging.h b/src/base/logging.h
index 889c6885b2..a9fd0025ba 100644
--- a/src/base/logging.h
+++ b/src/base/logging.h
@@ -70,7 +70,7 @@ V8_BASE_EXPORT void SetDcheckFunction(void (*dcheck_Function)(const char*, int,
       V8_Dcheck(__FILE__, __LINE__, message); \
     }                                         \
   } while (0)
-#define DCHECK(condition) DCHECK_WITH_MSG(condition, #condition)
+#define DCHECK(condition) DCHECK_WITH_MSG((condition) || true, #condition)
 
 // Helper macro for binary operators.
 // Don't use this macro directly in your code, use CHECK_EQ et al below.
