diff --git a/src/base/logging.h b/src/base/logging.h
index e852dde8dd..a57ddf5404 100644
--- a/src/base/logging.h
+++ b/src/base/logging.h
@@ -62,7 +62,7 @@ DEFINE_CHECK_OP_IMPL(GT, > )
       V8_Fatal(__FILE__, __LINE__, "Debug check failed: %s.", message); \
     }                                                                   \
   } while (0)
-#define DCHECK(condition) DCHECK_WITH_MSG(condition, #condition)
+#define DCHECK(condition) DCHECK_WITH_MSG((condition) || true, #condition)
 
 // Helper macro for binary operators.
 // Don't use this macro directly in your code, use CHECK_EQ et al below.
