diff --git a/src/heap/heap.cc b/src/heap/heap.cc
index 04de6ea6b6..62a5856f34 100644
--- a/src/heap/heap.cc
+++ b/src/heap/heap.cc
@@ -1718,7 +1718,7 @@ void Heap::Scavenge() {
 
   if (mark_compact_collector()->sweeper().sweeping_in_progress() &&
       memory_allocator_->unmapper()->NumberOfDelayedChunks() >
-          kMaxSemiSpaceSizeInKB / Page::kPageSize) {
+          static_cast<int>(new_space_->MaximumCapacity() / Page::kPageSize)) {
     mark_compact_collector()->EnsureSweepingCompleted();
   }
 
