diff --git a/src/heap/heap.cc b/src/heap/heap.cc
index 2e6cfab292..61e15efd53 100644
--- a/src/heap/heap.cc
+++ b/src/heap/heap.cc
@@ -1885,6 +1885,12 @@ void Heap::Scavenge() {
   IncrementalMarking::PauseBlackAllocationScope pause_black_allocation(
       incremental_marking());
 
+  if (mark_compact_collector()->sweeper().sweeping_in_progress() &&
+      memory_allocator_->unmapper()->NumberOfDelayedChunks() >
+          kMaxSemiSpaceSizeInKB / Page::kPageSize) {
+    mark_compact_collector()->EnsureSweepingCompleted();
+  }
+
   mark_compact_collector()->sweeper().EnsureNewSpaceCompleted();
 
   SetGCState(SCAVENGE);
diff --git a/src/heap/spaces.h b/src/heap/spaces.h
index d5dc5b1115..230a127809 100644
--- a/src/heap/spaces.h
+++ b/src/heap/spaces.h
@@ -1199,6 +1199,11 @@ class V8_EXPORT_PRIVATE MemoryAllocator {
 
     bool has_delayed_chunks() { return delayed_regular_chunks_.size() > 0; }
 
+    int NumberOfDelayedChunks() {
+      base::LockGuard<base::Mutex> guard(&mutex_);
+      return static_cast<int>(delayed_regular_chunks_.size());
+    }
+
    private:
     static const int kReservedQueueingSlots = 64;
     static const int kMaxUnmapperTasks = 24;
