diff --git a/src/builtins/builtins-promise-gen.cc b/src/builtins/builtins-promise-gen.cc
index d6d4772fd3..1332a4f9d0 100644
--- a/src/builtins/builtins-promise-gen.cc
+++ b/src/builtins/builtins-promise-gen.cc
@@ -2063,6 +2063,10 @@ TF_BUILTIN(PromiseAllResolveElementClosure, PromiseBuiltinsAssembler) {
   {
     VARIABLE(var_elements, MachineRepresentation::kTagged,
              LoadElements(values_array));
+    // Check that the {values_array} is still in fast mode.
+    Node* const elements_kind = LoadMapElementsKind(LoadMap(values_array));
+    GotoIf(Word32Equal(elements_kind, Int32Constant(DICTIONARY_ELEMENTS)),
+           &runtime_set_element);
     PossiblyGrowElementsCapacity(SMI_PARAMETERS, PACKED_ELEMENTS, values_array,
                                  index, &var_elements, SmiConstant(1),
                                  &runtime_set_element);
