diff --git a/BUILD.gn b/BUILD.gn
index 494ba22..894797c 100644
--- a/BUILD.gn
+++ b/BUILD.gn
@@ -907,9 +907,19 @@ if (v8_use_external_startup_data) {
     ]
     public_deps = [
       ":natives_blob",
-      ":run_mksnapshot",
     ]
 
+    if (v8_snapshot_toolchain == "//build/toolchain/linux:clang_arm" ||
+        v8_snapshot_toolchain == "//build/toolchain/linux:clang_arm64") {
+      public_deps += [
+        ":mksnapshot($v8_snapshot_toolchain)",
+      ]
+    } else {
+      public_deps += [
+        ":run_mksnapshot",
+      ]
+    }
+
     sources = [
       "src/setup-isolate-deserialize.cc",
       "src/snapshot/natives-external.cc",
