diff --git a/test/message/testcfg.py b/test/message/testcfg.py
index 0576458af2..1bbb2b16d8 100644
--- a/test/message/testcfg.py
+++ b/test/message/testcfg.py
@@ -107,7 +107,7 @@ class MessageTestSuite(testsuite.TestSuite):
         expected_lines, actual_lines, fillvalue=''):
       pattern = re.escape(expected.rstrip() % env)
       pattern = pattern.replace("\\*", ".*")
-      pattern = pattern.replace("\\{NUMBER\\}", "\d(?:\.\d*)?")
+      pattern = pattern.replace("\\{NUMBER\\}", "\d+(?:\.\d*)?")
       pattern = "^%s$" % pattern
       if not re.match(pattern, actual):
         return True
