diff --git a/BUILD.gn b/BUILD.gn
index 8587356..b9f0736 100644
--- a/BUILD.gn
+++ b/BUILD.gn
@@ -113,7 +113,7 @@ config("internal_config") {
 
   include_dirs = [ "." ]
 
-  if (is_component_build) {
+  if (is_component_build || is_electron_build) {
     defines = [ "BUILDING_V8_SHARED" ]
   }
 }
@@ -127,14 +127,14 @@ config("internal_config_base") {
 # This config should be applied to code using the libplatform.
 config("libplatform_config") {
   include_dirs = [ "include" ]
-  if (is_component_build) {
+  if (is_component_build || is_electron_build) {
     defines = [ "USING_V8_PLATFORM_SHARED" ]
   }
 }
 
 # This config should be applied to code using the libbase.
 config("libbase_config") {
-  if (is_component_build) {
+  if (is_component_build || is_electron_build) {
     defines = [ "USING_V8_BASE_SHARED" ]
   }
   libs = []
@@ -151,7 +151,7 @@ config("libsampler_config") {
 # This config should only be applied to code using V8 and not any V8 code
 # itself.
 config("external_config") {
-  if (is_component_build) {
+  if (is_component_build || is_electron_build) {
     defines = [ "USING_V8_SHARED" ]
   }
   include_dirs = [ "include" ]
@@ -2265,7 +2265,7 @@ v8_component("v8_libbase") {
 
   defines = []
 
-  if (is_component_build) {
+  if (is_component_build || is_electron_build) {
     defines = [ "BUILDING_V8_BASE_SHARED" ]
   }
 
@@ -2355,7 +2355,7 @@ v8_component("v8_libplatform") {
 
   configs = [ ":internal_config_base" ]
 
-  if (is_component_build) {
+  if (is_component_build || is_electron_build) {
     defines = [ "BUILDING_V8_PLATFORM_SHARED" ]
   }
 
@@ -2423,6 +2423,8 @@ if (current_toolchain == v8_snapshot_toolchain) {
 
     configs = [ ":internal_config" ]
 
+    configs += [ "//chromiumcontent:build_time_executable" ]
+
     deps = [
       ":v8_base",
       ":v8_libbase",
@@ -2498,7 +2500,7 @@ group("gn_all") {
   }
 }
 
-if (is_component_build) {
+if (is_component_build || is_electron_build) {
   v8_component("v8") {
     sources = [
       "src/v8dll-main.cc",
diff --git a/gni/v8.gni b/gni/v8.gni
index 3759572..9a3195f 100644
--- a/gni/v8.gni
+++ b/gni/v8.gni
@@ -4,6 +4,7 @@
 
 import("//build/config/sanitizers/sanitizers.gni")
 import("//build/config/v8_target_cpu.gni")
+import("//chromiumcontent/config.gni")
 
 declare_args() {
   # Indicate if valgrind was fetched as a custom deps to make it available on
@@ -113,7 +114,7 @@ template("v8_executable") {
 }
 
 template("v8_component") {
-  component(target_name) {
+  target(component_electron, target_name) {
     forward_variables_from(invoker, "*", [ "configs" ])
     configs += invoker.configs
     configs -= v8_remove_configs
