diff --git a/BUILD.gn b/BUILD.gn
index e3d12f49..9c64371b 100644
--- a/BUILD.gn
+++ b/BUILD.gn
@@ -3,6 +3,7 @@
 # found in the LICENSE file.
 
 import("//build/config/host_byteorder.gni")
+import("//chromiumcontent/config.gni")
 import("//third_party/icu/config.gni")
 
 if (is_android) {
@@ -33,7 +34,7 @@ config("icu_config") {
     "U_ENABLE_DYLOAD=0",
   ]
 
-  if (!is_component_build) {
+  if (!is_component_build && !is_electron_build) {
     defines += [ "U_STATIC_IMPLEMENTATION" ]
   }
 
@@ -100,7 +101,7 @@ config("icu_code") {
   }
 }
 
-component("icui18n") {
+target(component_electron, "icui18n") {
   # find  source/i18n -maxdepth 1  ! -type d  | egrep  '\.(c|cpp|h)$' |\
   # sort | sed 's/^\(.*\)$/    "\1",/'
   sources = [
@@ -524,6 +525,9 @@ component("icui18n") {
   ]
 
   defines = [ "U_I18N_IMPLEMENTATION" ]
+  if (is_electron_build) {
+    defines += [ "U_COMBINED_IMPLEMENTATION" ]
+  }
   deps = [
     ":icuuc",
   ]
@@ -542,7 +546,7 @@ component("icui18n") {
   public_configs = [ ":icu_config" ]
 }
 
-component("icuuc") {
+target(component_electron, "icuuc") {
   # find  source/common -maxdepth 1  ! -type d  | egrep  '\.(c|cpp|h)$' |\
   # sort | sed 's/^\(.*\)$/    "\1",/'
   sources = [
@@ -927,6 +931,9 @@ component("icuuc") {
     "source/common/unicode/uversion.h",
   ]
   defines = [ "U_COMMON_IMPLEMENTATION" ]
+  if (is_electron_build) {
+    defines += [ "U_COMBINED_IMPLEMENTATION" ]
+  }
   deps = [
     ":icudata",
   ]
