diff --git a/source/common/unicode/unistr.h b/source/common/unicode/unistr.h
index 6f62244..29b6b91 100644
--- a/source/common/unicode/unistr.h
+++ b/source/common/unicode/unistr.h
@@ -2960,7 +2960,7 @@ public:
   /** Construct an empty UnicodeString.
    * @stable ICU 2.0
    */
-  inline UnicodeString();
+  UnicodeString();
 
   /**
    * Construct a UnicodeString with capacity to hold <TT>capacity</TT> UChars
@@ -3765,15 +3765,6 @@ UnicodeString::getArrayStart() const {
 }
 
 //========================================
-// Default constructor
-//========================================
-
-inline
-UnicodeString::UnicodeString() {
-  fUnion.fStackFields.fLengthAndFlags=kShortString;
-}
-
-//========================================
 // Read-only implementation methods
 //========================================
 inline UBool
diff --git a/source/common/unistr.cpp b/source/common/unistr.cpp
index bdd58ec..60f6835 100644
--- a/source/common/unistr.cpp
+++ b/source/common/unistr.cpp
@@ -146,7 +146,9 @@ UnicodeString::releaseArray() {
 // Constructors
 //========================================
 
-// The default constructor is inline in unistr.h.
+UnicodeString::UnicodeString() {
+  fUnion.fStackFields.fLengthAndFlags=kShortString;
+}
 
 UnicodeString::UnicodeString(int32_t capacity, UChar32 c, int32_t count) {
   fUnion.fFields.fLengthAndFlags = 0;
