diff --git a/BUILD.gn b/BUILD.gn
index 1dd7c61d..80068141 100644
--- a/BUILD.gn
+++ b/BUILD.gn
@@ -2,6 +2,7 @@
 # Use of this source code is governed by a BSD-style license that can be
 # found in the LICENSE file.
 
+import("//chromiumcontent/config.gni")
 import("//third_party/icu/config.gni")
 
 if (is_android) {
@@ -38,7 +39,7 @@ config("icu_config") {
     "U_NOEXCEPT=",
   ]
 
-  if (!is_component_build) {
+  if (!is_component_build && !is_electron_build) {
     defines += [ "U_STATIC_IMPLEMENTATION" ]
   }
 
@@ -122,7 +123,7 @@ config("icu_code") {
   }
 }
 
-component("icui18n") {
+target(component_electron, "icui18n") {
   # find  source/i18n -maxdepth 1  ! -type d  | egrep  '\.(c|cpp|h)$' |\
   # sort | sed 's/^\(.*\)$/    "\1",/'
   sources = [
@@ -553,6 +554,9 @@ component("icui18n") {
   ]
 
   defines = [ "U_I18N_IMPLEMENTATION" ]
+  if (is_electron_build) {
+    defines += [ "U_COMBINED_IMPLEMENTATION" ]
+  }
   deps = [
     ":icuuc",
   ]
@@ -579,7 +583,7 @@ component("icui18n") {
   }
 }
 
-component("icuuc") {
+target(component_electron, "icuuc") {
   # find  source/common -maxdepth 1  ! -type d  | egrep  '\.(c|cpp|h)$' |\
   # sort | sed 's/^\(.*\)$/    "\1",/'
   sources = [
@@ -954,6 +958,9 @@ component("icuuc") {
     "source/common/unicode/uversion.h",
   ]
   defines = [ "U_COMMON_IMPLEMENTATION" ]
+  if (is_electron_build) {
+    defines += [ "U_COMBINED_IMPLEMENTATION" ]
+  }
   deps = [
     ":icudata",
   ]
