diff --git a/source/common/unicode/unistr.h b/source/common/unicode/unistr.h
index 63f1933..44eeecd 100644
--- a/source/common/unicode/unistr.h
+++ b/source/common/unicode/unistr.h
@@ -2959,7 +2959,7 @@ public:
   /** Construct an empty UnicodeString.
    * @stable ICU 2.0
    */
-  inline UnicodeString();
+  UnicodeString();
 
   /**
    * Construct a UnicodeString with capacity to hold <TT>capacity</TT> UChars
@@ -3767,15 +3767,6 @@ UnicodeString::getArrayStart() const {
 }
 
 //========================================
-// Default constructor
-//========================================
-
-inline
-UnicodeString::UnicodeString() {
-  fUnion.fStackFields.fLengthAndFlags=kShortString;
-}
-
-//========================================
 // Read-only implementation methods
 //========================================
 inline UBool
diff --git a/source/common/unistr.cpp b/source/common/unistr.cpp
index f957c53..a70d0b6 100644
--- a/source/common/unistr.cpp
+++ b/source/common/unistr.cpp
@@ -144,7 +144,9 @@ UnicodeString::releaseArray() {
 // Constructors
 //========================================
 
-// The default constructor is inline in unistr.h.
+UnicodeString::UnicodeString() {
+  fUnion.fStackFields.fLengthAndFlags=kShortString;
+}
 
 UnicodeString::UnicodeString(int32_t capacity, UChar32 c, int32_t count) {
   fUnion.fFields.fLengthAndFlags = 0;
