diff --git a/BUILD.gn b/BUILD.gn
index df54a977..38a4956 100644
--- a/BUILD.gn
+++ b/BUILD.gn
@@ -688,10 +688,10 @@ group("gn_all") {
       if (!is_android) {
         deps += [ "//chrome/test:load_library_perf_tests" ]
       }
-      deps += [
-        "//native_client/src/trusted/platform_qualify:vcpuid",
-        "//third_party/libjpeg_turbo:simd_asm",
-      ]
+      if (enable_nacl) {
+        deps += [ "//native_client/src/trusted/platform_qualify:vcpuid" ]
+      }
+      deps += [ "//third_party/libjpeg_turbo:simd_asm" ]
     }
     if (is_linux && current_toolchain == host_toolchain) {
       deps += [ "//v8:v8_shell" ]
