diff --git a/base/process/process_metrics_win.cc b/base/process/process_metrics_win.cc
index 61f0bf4ad06f..897ec93a16a6 100644
--- a/base/process/process_metrics_win.cc
+++ b/base/process/process_metrics_win.cc
@@ -325,11 +325,11 @@ bool ProcessMetrics::GetIOCounters(IoCounters* io_counters) const {
 
 ProcessMetrics::ProcessMetrics(ProcessHandle process) : last_system_time_(0) {
   if (process) {
-    HANDLE duplicate_handle;
+    HANDLE duplicate_handle = INVALID_HANDLE_VALUE;
     BOOL result = ::DuplicateHandle(::GetCurrentProcess(), process,
                                     ::GetCurrentProcess(), &duplicate_handle,
                                     PROCESS_QUERY_INFORMATION, FALSE, 0);
-    DCHECK(result);
+    DPCHECK(result);
     process_.Set(duplicate_handle);
   }
 }