diff --git a/content/browser/service_worker/service_worker_version.cc b/content/browser/service_worker/service_worker_version.cc
index 2417abcfdae2..23f0f0920eed 100644
--- a/content/browser/service_worker/service_worker_version.cc
+++ b/content/browser/service_worker/service_worker_version.cc
@@ -54,10 +54,10 @@

 namespace content {

-using StatusCallback = ServiceWorkerVersion::StatusCallback;
-
 namespace {

+using StatusCallback = ServiceWorkerVersion::StatusCallback;
+
 // Timeout for an installed worker to start.
 constexpr base::TimeDelta kStartInstalledWorkerTimeout =
     base::TimeDelta::FromSeconds(60);
