diff --git a/third_party/WebKit/Source/platform/graphics/compositing/ContentLayerClientImpl.cpp b/third_party/WebKit/Source/platform/graphics/compositing/ContentLayerClientImpl.cpp
index 0d894db55e4f..7565fc85f551 100644
--- a/third_party/WebKit/Source/platform/graphics/compositing/ContentLayerClientImpl.cpp
+++ b/third_party/WebKit/Source/platform/graphics/compositing/ContentLayerClientImpl.cpp
@@ -264,8 +264,10 @@ scoped_refptr<cc::PictureLayer> ContentLayerClientImpl::UpdateCcPictureLayer(
                        chunk->begin_index, chunk->end_index,
                        DisplayItemList::kSkipNonDrawings |
                            DisplayItemList::kShownOnlyDisplayItemTypes));
+#if DCHECK_IS_ON()
     json->SetString("propertyTreeState",
                     chunk->properties.property_tree_state.ToTreeString());
+#endif  // DCHECK_IS_ON()
     paint_chunk_debug_data_->PushObject(std::move(json));
   }
 #endif
