diff --git a/chrome/BUILD.gn b/chrome/BUILD.gn
index fbcf20c..78789a9 100644
--- a/chrome/BUILD.gn
+++ b/chrome/BUILD.gn
@@ -1656,6 +1656,10 @@ if (is_chrome_branded && !is_android) {
   }
 }
 
+if (is_electron_gn_build) {
+  group("packed_resources") {
+  }
+} else {
 chrome_paks("packed_resources") {
   if (is_mac) {
     output_dir = "$root_gen_dir/repack"
@@ -1677,6 +1681,7 @@ chrome_paks("packed_resources") {
     ]
   }
 }
+}
 
 repack("browser_tests_pak") {
   sources = [
