diff --git a/base/strings/string_piece.h b/base/strings/string_piece.h
index 8a7d0d8710b4..beb55792af6d 100644
--- a/base/strings/string_piece.h
+++ b/base/strings/string_piece.h
@@ -220,7 +220,7 @@ template <typename STRING_TYPE> class BasicStringPiece {
   }

   constexpr value_type operator[](size_type i) const {
-    CHECK(i < length_);
+    // CHECK(i < length_);
     return ptr_[i];
   }

@@ -235,13 +235,13 @@ template <typename STRING_TYPE> class BasicStringPiece {
   }

   constexpr void remove_prefix(size_type n) {
-    CHECK(n <= length_);
+    // CHECK(n <= length_);
     ptr_ += n;
     length_ -= n;
   }

   constexpr void remove_suffix(size_type n) {
-    CHECK(n <= length_);
+    // CHECK(n <= length_);
     length_ -= n;
   }
